// @gits-id/pagination/dist/pagination.mjs
import { defineComponent as E, toRefs as I, ref as D, computed as n, watch as C, openBlock as r, createElementBlock as o, Fragment as k, renderList as R, unref as e, normalizeClass as u, withModifiers as p, renderSlot as g, toDisplayString as P, createCommentVNode as j, createElementVNode as s } from "vue";
import q from "pagino";
const A = {
  class: "relative z-0 inline-flex rounded-md shadow-sm -space-x-px",
  "aria-label": "Pagination"
}, G = ["disabled"], H = /* @__PURE__ */ s("span", { class: "sr-only" }, "Previous", -1), J = /* @__PURE__ */ s("svg", {
  class: "h-5 w-5",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  "aria-hidden": "true"
}, [
  /* @__PURE__ */ s("path", {
    "fill-rule": "evenodd",
    d: "M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z",
    "clip-rule": "evenodd"
  })
], -1), K = ["disabled"], O = /* @__PURE__ */ s("span", { class: "sr-only" }, "Next", -1), Q = /* @__PURE__ */ s("svg", {
  class: "h-5 w-5",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  "aria-hidden": "true"
}, [
  /* @__PURE__ */ s("path", {
    "fill-rule": "evenodd",
    d: "M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z",
    "clip-rule": "evenodd"
  })
], -1), T = ["disabled"], U = /* @__PURE__ */ s("span", { class: "sr-only" }, "First", -1), W = /* @__PURE__ */ s("svg", {
  class: "h-4 w-4",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor"
}, [
  /* @__PURE__ */ s("path", {
    "fill-rule": "evenodd",
    d: "M15.707 15.707a1 1 0 01-1.414 0l-5-5a1 1 0 010-1.414l5-5a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 010 1.414zm-6 0a1 1 0 01-1.414 0l-5-5a1 1 0 010-1.414l5-5a1 1 0 011.414 1.414L5.414 10l4.293 4.293a1 1 0 010 1.414z",
    "clip-rule": "evenodd"
  })
], -1), X = ["disabled"], Y = /* @__PURE__ */ s("span", { class: "sr-only" }, "Last", -1), Z = /* @__PURE__ */ s("svg", {
  class: "h-4 w-4",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor"
}, [
  /* @__PURE__ */ s("path", {
    "fill-rule": "evenodd",
    d: "M10.293 15.707a1 1 0 010-1.414L14.586 10l-4.293-4.293a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z",
    "clip-rule": "evenodd"
  }),
  /* @__PURE__ */ s("path", {
    "fill-rule": "evenodd",
    d: "M4.293 15.707a1 1 0 010-1.414L8.586 10 4.293 5.707a1 1 0 011.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z",
    "clip-rule": "evenodd"
  })
], -1), ee = {
  key: 4,
  class: "relative inline-flex items-center px-4 py-2 border border-gray-300 bg-white text-sm font-medium text-gray-700"
}, te = ["onClick"], se = {
  key: 6,
  class: "relative inline-flex items-center px-4 py-2 border border-gray-300 text-sm font-medium hover:bg-gray-50"
}, re = /* @__PURE__ */ E({
  __name: "VPagination",
  props: {
    modelValue: {
      type: Number,
      default: 1
    },
    showFirst: {
      type: Boolean,
      default: !0
    },
    showPrevious: {
      type: Boolean,
      default: !0
    },
    showNext: {
      type: Boolean,
      default: !0
    },
    showLast: {
      type: Boolean,
      default: !0
    },
    totalItems: {
      type: Number,
      default: 0
    },
    siblingCount: {
      type: Number,
      default: 1
    },
    boundaryCount: {
      type: Number,
      default: 1
    },
    itemsPerPage: {
      type: Number,
      default: 10
    },
    activeClass: {
      type: String,
      default: "bg-primary-600 border-primary-600 hover:!bg-primary-700 text-white !font-semibold"
    },
    defaultClass: {
      type: String,
      default: "text-gray-700"
    },
    simple: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(N, { emit: L }) {
    const h = N, {
      totalItems: B,
      modelValue: x,
      itemsPerPage: $,
      activeClass: z,
      defaultClass: V,
      simple: w
    } = I(h), i = D(x.value), c = n(
      () => Math.ceil(B.value / $.value)
    ), a = n(() => new q({
      ...h,
      page: i.value,
      count: c.value,
      onChange: (t) => {
        Number.isNaN(t) || (i.value = t, _(t));
      }
    })), M = n(() => a.value.getPages()), m = n(() => i.value === 1), y = n(() => i.value === c.value), b = n(() => i.value >= c.value), f = n(() => i.value <= 1), F = n(() => z.value), S = n(() => V.value), _ = (t) => {
      L("update:modelValue", t);
    };
    return C(
      i,
      (t) => {
        _(t);
      },
      { immediate: !0 }
    ), C(
      x,
      (t) => {
        i.value = t;
      },
      { immediate: !0 }
    ), (t, d) => (r(), o("nav", A, [
      (r(!0), o(k, null, R(e(M), (l) => (r(), o(k, { key: l }, [
        l === "previous" ? (r(), o("button", {
          key: 0,
          disabled: e(f),
          class: u(["relative inline-flex items-center px-2 py-2 border border-gray-300 bg-white text-sm font-medium", {
            "rounded-l-md": !e(a).showFirst,
            "cursor-not-allowed text-gray-300": e(f),
            "text-gray-600 hover:bg-gray-50": !e(f)
          }]),
          onClick: d[0] || (d[0] = p((v) => e(a).previous(), ["prevent"]))
        }, [
          g(t.$slots, "btnPrev", {}, () => [
            H,
            J
          ])
        ], 10, G)) : l === "next" ? (r(), o("button", {
          key: 1,
          disabled: e(b),
          class: u(["relative inline-flex items-center px-2 py-2 border border-gray-300 bg-white text-sm font-medium", {
            "rounded-r-md": !e(a).showLast,
            "cursor-not-allowed text-gray-300": e(b),
            "text-gray-600 hover:bg-gray-50": !e(b)
          }]),
          onClick: d[1] || (d[1] = p((v) => e(a).next(), ["prevent"]))
        }, [
          g(t.$slots, "btnNext", {}, () => [
            O,
            Q
          ])
        ], 10, K)) : l === "first" ? (r(), o("button", {
          key: 2,
          disabled: e(m),
          class: u(["relative inline-flex items-center px-2 py-2 rounded-l-md border border-gray-300 bg-white text-sm font-medium", {
            "cursor-not-allowed text-gray-300": e(m),
            "text-gray-600 hover:bg-gray-50": !e(m)
          }]),
          onClick: d[2] || (d[2] = p((v) => e(a).first(), ["prevent"]))
        }, [
          g(t.$slots, "btnFirst", {}, () => [
            U,
            W
          ])
        ], 10, T)) : l === "last" ? (r(), o("button", {
          key: 3,
          disabled: e(y),
          class: u(["relative inline-flex items-center px-2 py-2 rounded-r-md border border-gray-300 bg-white text-sm font-medium", {
            "cursor-not-allowed text-gray-300": e(y),
            "text-gray-600 hover:bg-gray-50": !e(y)
          }]),
          onClick: d[3] || (d[3] = p((v) => e(a).last(), ["prevent"]))
        }, [
          g(t.$slots, "btnLast", {}, () => [
            Y,
            Z
          ])
        ], 10, X)) : l === "start-ellipsis" || l === "end-ellipsis" ? (r(), o("span", ee, " ... ")) : e(w) ? e(w) && e(a).page === l ? (r(), o("span", se, P(l), 1)) : j("", !0) : (r(), o("a", {
          key: 5,
          href: "#",
          class: u(["relative inline-flex items-center px-4 py-2 border border-gray-300 text-sm font-medium hover:bg-gray-50", e(a).page === l ? e(F) : e(S)]),
          onClick: p((v) => e(a).setPage(l), ["prevent"])
        }, P(l), 11, te))
      ], 64))), 128))
    ]));
  }
});
export {
  re as VPagination,
  re as default
};
