// @gits-id/select/dist/select.mjs
import { defineComponent as N, computed as S, openBlock as h, createElementBlock as w, normalizeClass as I, unref as s, createElementVNode as m, toRefs as fe, createBlock as P, resolveDynamicComponent as gt, mergeProps as $, withCtx as A, renderSlot as R, createVNode as O, toDisplayString as C, createCommentVNode as B, h as ht, ref as ie, watch as J, createTextVNode as q, normalizeStyle as qe, Fragment as He, onMounted as yt, onUnmounted as bt, normalizeProps as mt, withDirectives as vt, isRef as Qe, vModelDynamic as xt, toHandlers as wt, Transition as kt, renderList as Ct } from "vue";
import { Listbox as St, ListboxLabel as Tt, ListboxButton as It, ListboxOptions as Bt, ListboxOption as jt } from "@headlessui/vue";
import { PlusIcon as Lt, TrashIcon as Ft, CameraIcon as Rt, XIcon as Vt, ChevronDownIcon as Ot, CheckIcon as At } from "@heroicons/vue/solid";
import { getBgColor as Et } from "@gits-id/utils";
import { useField as Be, ErrorMessage as Pt } from "vee-validate";
import _t from "@gits-id/tooltip";
const Mt = /* @__PURE__ */ m("circle", {
  class: "opacity-25",
  cx: "12",
  cy: "12",
  r: "10",
  stroke: "currentColor",
  "stroke-width": "4"
}, null, -1), Nt = /* @__PURE__ */ m("path", {
  class: "opacity-75",
  fill: "currentColor",
  d: "M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
}, null, -1), Ut = [
  Mt,
  Nt
], Ke = /* @__PURE__ */ N({
  __name: "VSpinner",
  props: {
    xLarge: {
      type: Boolean,
      default: !1
    },
    large: {
      type: Boolean,
      default: !1
    },
    small: {
      type: Boolean,
      default: !1
    },
    xSmall: {
      type: Boolean,
      default: !1
    },
    color: {
      type: String,
      default: ""
    }
  },
  setup(e) {
    const t = e, r = S(() => t.large ? "h-8 w-8" : t.small ? "h-4 w-4" : t.xLarge ? "h-10 w-10" : t.xSmall ? "h-3 w-3" : "h-5 w-5"), n = S(() => {
      switch (t.color) {
        case "primary":
          return "text-primary-600";
        case "info":
          return "text-info-600";
        case "warning":
          return "text-warning-600";
        case "error":
          return "text-error-600";
        case "success":
          return "text-success-600";
        case "white":
          return "text-white-600";
        default:
          return t.color || "text-gray-600";
      }
    });
    return (o, a) => (h(), w("svg", {
      class: I(["animate-spin inline", [s(r), s(n)]]),
      xmlns: "http://www.w3.org/2000/svg",
      fill: "none",
      viewBox: "0 0 24 24"
    }, Ut, 2));
  }
}), Dt = {
  key: 0,
  class: "ml-2"
}, W = /* @__PURE__ */ N({
  __name: "VBtn",
  props: {
    size: {
      type: String,
      default: "md",
      validator: (e) => ["xs", "sm", "md", "lg", "xl"].includes(e)
    },
    color: {
      type: String,
      default: "default"
    },
    to: {
      type: [String, Object],
      default: ""
    },
    href: {
      type: String,
      default: ""
    },
    text: {
      type: Boolean,
      default: !1
    },
    outlined: {
      type: Boolean,
      default: !1
    },
    rounded: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    ring: {
      type: Boolean,
      default: !0
    },
    tile: {
      type: Boolean,
      default: !1
    },
    icon: {
      type: Boolean,
      default: !1
    },
    block: {
      type: Boolean,
      default: !1
    },
    type: {
      type: String,
      default: "button"
    },
    solid: {
      type: Boolean,
      default: !1
    },
    noRing: {
      type: Boolean,
      default: !1
    },
    loading: {
      type: Boolean,
      default: !1
    },
    loadingText: {
      type: String,
      default: ""
    },
    newTab: {
      type: Boolean,
      default: !1
    },
    shadow: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, { disabled: r, loading: n, loadingText: o } = fe(t), a = S(() => t.to ? "router-link" : t.href ? "a" : "button"), l = S(() => {
      const x = {};
      return t.to ? x.to = t.to : t.href ? x.href = t.href : x.type = t.type, x;
    }), i = S(() => {
      switch (t.color) {
        case "primary":
        case "info":
        case "warning":
        case "error":
        case "success":
          return "white";
        case "white":
          return "black";
        default:
          return t.color;
      }
    }), c = {
      default: "border-gray-400 text-gray-700 bg-white hover:border-gray-500 active:bg-gray-50 active:border-gray-600 active:text-gray-600",
      primary: "border-primary-500 bg-primary-500 text-white hover:bg-primary-600 hover:border-primary-600 active:bg-primary-400 active:border-primary-400",
      secondary: "border-secondary-500 bg-secondary-500 text-white hover:bg-secondary-600 hover:border-secondary-600 active:bg-secondary-400 active:border-secondary-400",
      info: "border-info-500 bg-info-500 text-white hover:bg-info-600 hover:border-info-600 active:bg-info-400 active:border-info-400",
      warning: "border-warning-500 bg-warning-500 text-white hover:bg-warning-600 hover:border-warning-600 active:bg-warning-400 active:border-warning-400",
      success: "border-success-500 bg-success-500 text-white hover:bg-success-600 hover:border-success-600 active:bg-success-400 active:border-success-400",
      error: "border-error-500 bg-error-500 text-white hover:bg-error-600 hover:border-error-600 active:bg-error-400 active:border-error-400",
      dark: "border-gray-800 bg-gray-800 text-white hover:bg-gray-900 hover:border-gray-900 active:bg-gray-700 active:border-gray-700"
    }, u = {
      default: "border-gray-400 text-gray-700 hover:bg-gray-700 hover:border-gray-700 hover:text-white active:bg-gray-50 active:border-gray-600 active:text-gray-600",
      primary: "border-primary-500 bg-transparent text-primary-500 hover:bg-primary-600 hover:border-primary-600 hover:text-white active:bg-primary-400 active:border-primary-400",
      secondary: "border-secondary-500 bg-transparent text-secondary-500 hover:bg-secondary-600 hover:border-secondary-600 hover:text-white active:bg-secondary-400 active:border-secondary-400",
      info: "border-info-500 bg-transparent text-info-500 hover:bg-info-600 hover:border-info-600 hover:text-white active:bg-info-400 active:border-info-400",
      warning: "border-warning-500 bg-transparent text-warning-500 hover:bg-warning-600 hover:border-warning-600 hover:text-white active:bg-warning-400 active:border-warning-400",
      success: "border-success-500 bg-transparent text-success-500 hover:bg-success-600 hover:border-success-600 hover:text-white active:bg-success-400 active:border-success-400",
      error: "border-error-500 bg-transparent text-error-500 hover:bg-error-600 hover:border-error-600 hover:text-white active:bg-error-400 active:border-error-400",
      dark: "border-gray-800 bg-transparent text-gray-800 hover:bg-gray-900 hover:border-gray-900 hover:text-white active:bg-gray-700 active:border-gray-700"
    }, d = {
      default: "border-transparent bg-transparent text-gray-700 hover:text-gray-900 active:text-gray-600",
      primary: "border-transparent bg-transparent text-primary-500 hover:text-primary-600 active:text-primary-400",
      secondary: "border-transparent bg-transparent text-secondary-500 hover:text-secondary-600 active:text-secondary-400",
      info: "border-transparent bg-transparent text-info-500 hover:text-info-600 active:text-info-400",
      warning: "border-transparent bg-transparent text-warning-500 hover:text-warning-600 active:text-warning-400",
      success: "border-transparent bg-transparent text-success-500 hover:text-success-600 active:text-success-400",
      error: "border-transparent bg-transparent text-error-500 hover:text-error-600 active:text-error-400",
      dark: "border-transparent bg-transparent text-gray-800 hover:text-gray-900 active:text-gray-600"
    }, f = {
      xs: "px-2 py-1 text-xs",
      sm: "px-3 py-1.5 text-sm",
      md: "px-4 py-2 text-base",
      lg: "px-5 py-3 text-lg",
      xl: "px-6 py-4 text-xl"
    }, p = {
      default: "",
      primary: "focus:ring-primary-300",
      secondary: "focus:ring-secondary-300",
      info: "focus:ring-info-300",
      warning: "focus:ring-warning-300",
      success: "focus:ring-success-300",
      error: "focus:ring-error-300",
      dark: "focus:ring-gray-300"
    }, g = S(() => {
      let x = c[t.color];
      return t.outlined ? x = u[t.color] : t.text && (x = d[t.color]), x;
    }), T = S(() => {
      let x = "disabled:cursor-not-allowed disabled:shadow-none disabled:text-gray-400";
      t.outlined ? x += " disabled:border-gray-300 disabled:hover:bg-transparent disabled:hover:text-gray-400" : x += " disabled:bg-gray-300 disabled:border-gray-300";
      const v = {
        default: "disabled:hover:bg-transparent disabled:active:bg-transparent",
        primary: "disabled:hover:bg-primary-500 disabled:hover:border-primary-500 disabled:active:bg-primary-500",
        secondary: "disabled:hover:bg-secondary-500 disabled:hover:border-secondary-500 disabled:active:bg-secondary-500",
        info: "disabled:hover:bg-info-500 disabled:hover:border-info-500 disabled:active:bg-info-500",
        warning: "disabled:hover:bg-warning-500 disabled:hover:border-warning-500 disabled:active:bg-warning-500",
        error: "disabled:hover:bg-error-500 disabled:hover:border-error-500 disabled:active:bg-error-500",
        success: "disabled:hover:bg-success-500 disabled:hover:border-success-500 disabled:active:bg-success-500",
        dark: "disabled:hover:bg-gray-800 disabled:hover:border-gray-800 disabled:active:bg-gray-800"
      };
      return t.loading ? [
        "disabled:cursor-not-allowed disabled:shadow-none",
        v[t.color]
      ] : x;
    }), y = S(() => {
      const x = {
        xs: "w-[26px] h-[26px]",
        sm: "w-[34px] h-[34px]",
        md: "w-[42px] h-[42px]",
        lg: "w-[54px] h-[54px]",
        xl: "w-[62px] h-[62px]"
      }, v = {
        xs: "px-1",
        sm: "px-1.5",
        md: "px-2",
        lg: "px-2.5",
        xl: "px-3"
      };
      return [
        x[t.size],
        t.text ? "p-0" : v[t.size]
      ].join(" ");
    }), j = S(() => [
      "inline-flex justify-center items-center border rounded-md font-medium transition duration-300",
      T.value,
      g.value,
      {
        "focus:ring focus:ring-offset-2 focus:outline-none": !t.noRing,
        [p[t.color]]: !t.noRing,
        "rounded-full": t.rounded,
        "rounded-none": t.tile,
        shadow: t.shadow,
        "w-full": t.block,
        "btn-loading": t.loading,
        [f[t.size]]: !t.icon,
        [y.value]: t.icon
      }
    ]);
    return (x, v) => (h(), P(gt(s(a)), $({
      class: s(j),
      rel: e.newTab ? "noopener" : "",
      target: e.newTab ? "_blank" : "",
      disabled: s(r) || s(n)
    }, { ...s(l), ...x.$attrs }), {
      default: A(() => [
        s(n) ? R(x.$slots, "loading", { key: 0 }, () => [
          O(s(Ke), { color: s(i) }, null, 8, ["color"]),
          s(o) ? (h(), w("span", Dt, C(s(o)), 1)) : B("", !0)
        ]) : R(x.$slots, "default", { key: 1 })
      ]),
      _: 3
    }, 16, ["class", "rel", "target", "disabled"]));
  }
}), oe = /^[a-z0-9]+(-[a-z0-9]+)*$/, z = Object.freeze({
  left: 0,
  top: 0,
  width: 16,
  height: 16,
  rotate: 0,
  vFlip: !1,
  hFlip: !1
});
function pe(e) {
  return { ...z, ...e };
}
const ge = (e, t, r, n = "") => {
  const o = e.split(":");
  if (e.slice(0, 1) === "@") {
    if (o.length < 2 || o.length > 3)
      return null;
    n = o.shift().slice(1);
  }
  if (o.length > 3 || !o.length)
    return null;
  if (o.length > 1) {
    const i = o.pop(), c = o.pop(), u = {
      provider: o.length > 0 ? o[0] : n,
      prefix: c,
      name: i
    };
    return t && !ne(u) ? null : u;
  }
  const a = o[0], l = a.split("-");
  if (l.length > 1) {
    const i = {
      provider: n,
      prefix: l.shift(),
      name: l.join("-")
    };
    return t && !ne(i) ? null : i;
  }
  if (r && n === "") {
    const i = {
      provider: n,
      prefix: "",
      name: a
    };
    return t && !ne(i, r) ? null : i;
  }
  return null;
}, ne = (e, t) => e ? !!((e.provider === "" || e.provider.match(oe)) && (t && e.prefix === "" || e.prefix.match(oe)) && e.name.match(oe)) : !1;
function $t(e, t) {
  const r = { ...e };
  for (const n in z) {
    const o = n;
    if (t[o] !== void 0) {
      const a = t[o];
      if (r[o] === void 0) {
        r[o] = a;
        continue;
      }
      switch (o) {
        case "rotate":
          r[o] = (r[o] + a) % 4;
          break;
        case "hFlip":
        case "vFlip":
          r[o] = a !== r[o];
          break;
        default:
          r[o] = a;
      }
    }
  }
  return r;
}
function Ee(e, t, r = !1) {
  function n(a, l) {
    if (e.icons[a] !== void 0)
      return Object.assign({}, e.icons[a]);
    if (l > 5)
      return null;
    const i = e.aliases;
    if (i && i[a] !== void 0) {
      const u = i[a], d = n(u.parent, l + 1);
      return d && $t(d, u);
    }
    const c = e.chars;
    return !l && c && c[a] !== void 0 ? n(c[a], l + 1) : null;
  }
  const o = n(t, 0);
  if (o)
    for (const a in z)
      o[a] === void 0 && e[a] !== void 0 && (o[a] = e[a]);
  return o && r ? pe(o) : o;
}
function zt(e) {
  for (const t in z)
    if (e[t] !== void 0)
      return !0;
  return !1;
}
function Ye(e, t, r) {
  r = r || {};
  const n = [];
  if (typeof e != "object" || typeof e.icons != "object")
    return n;
  e.not_found instanceof Array && e.not_found.forEach((l) => {
    t(l, null), n.push(l);
  });
  const o = e.icons;
  Object.keys(o).forEach((l) => {
    const i = Ee(e, l, !0);
    i && (t(l, i), n.push(l));
  });
  const a = r.aliases || "all";
  if (a !== "none" && typeof e.aliases == "object") {
    const l = e.aliases;
    Object.keys(l).forEach((i) => {
      if (a === "variations" && zt(l[i]))
        return;
      const c = Ee(e, i, !0);
      c && (t(i, c), n.push(i));
    });
  }
  return n;
}
const ke = {
  provider: "string",
  aliases: "object",
  not_found: "object"
};
for (const e in z)
  ke[e] = typeof z[e];
function Ge(e) {
  if (typeof e != "object" || e === null)
    return null;
  const t = e;
  if (typeof t.prefix != "string" || !e.icons || typeof e.icons != "object")
    return null;
  for (const o in ke)
    if (e[o] !== void 0 && typeof e[o] !== ke[o])
      return null;
  const r = t.icons;
  for (const o in r) {
    const a = r[o];
    if (!o.match(oe) || typeof a.body != "string")
      return null;
    for (const l in z)
      if (a[l] !== void 0 && typeof a[l] != typeof z[l])
        return null;
  }
  const n = t.aliases;
  if (n)
    for (const o in n) {
      const a = n[o], l = a.parent;
      if (!o.match(oe) || typeof l != "string" || !r[l] && !n[l])
        return null;
      for (const i in z)
        if (a[i] !== void 0 && typeof a[i] != typeof z[i])
          return null;
    }
  return t;
}
const qt = 1;
let ce = /* @__PURE__ */ Object.create(null);
try {
  const e = window || self;
  e && e._iconifyStorage.version === qt && (ce = e._iconifyStorage.storage);
} catch {
}
function Ht(e, t) {
  return {
    provider: e,
    prefix: t,
    icons: /* @__PURE__ */ Object.create(null),
    missing: /* @__PURE__ */ Object.create(null)
  };
}
function Z(e, t) {
  ce[e] === void 0 && (ce[e] = /* @__PURE__ */ Object.create(null));
  const r = ce[e];
  return r[t] === void 0 && (r[t] = Ht(e, t)), r[t];
}
function je(e, t) {
  if (!Ge(t))
    return [];
  const r = Date.now();
  return Ye(t, (n, o) => {
    o ? e.icons[n] = o : e.missing[n] = r;
  });
}
function Qt(e, t, r) {
  try {
    if (typeof r.body == "string")
      return e.icons[t] = Object.freeze(pe(r)), !0;
  } catch {
  }
  return !1;
}
function Kt(e, t) {
  const r = e.icons[t];
  return r === void 0 ? null : r;
}
let ae = !1;
function Je(e) {
  return typeof e == "boolean" && (ae = e), ae;
}
function Yt(e) {
  const t = typeof e == "string" ? ge(e, !0, ae) : e;
  return t ? Kt(Z(t.provider, t.prefix), t.name) : null;
}
function Gt(e, t) {
  const r = ge(e, !0, ae);
  if (!r)
    return !1;
  const n = Z(r.provider, r.prefix);
  return Qt(n, r.name, t);
}
function Jt(e, t) {
  if (typeof e != "object")
    return !1;
  if (typeof t != "string" && (t = typeof e.provider == "string" ? e.provider : ""), ae && t === "" && (typeof e.prefix != "string" || e.prefix === "")) {
    let n = !1;
    return Ge(e) && (e.prefix = "", Ye(e, (o, a) => {
      a && Gt(o, a) && (n = !0);
    })), n;
  }
  if (typeof e.prefix != "string" || !ne({
    provider: t,
    prefix: e.prefix,
    name: "a"
  }))
    return !1;
  const r = Z(t, e.prefix);
  return !!je(r, e);
}
const Pe = Object.freeze({
  inline: !1,
  width: null,
  height: null,
  hAlign: "center",
  vAlign: "middle",
  slice: !1,
  hFlip: !1,
  vFlip: !1,
  rotate: 0
});
function Xt(e, t) {
  const r = {};
  for (const n in e) {
    const o = n;
    if (r[o] = e[o], t[o] === void 0)
      continue;
    const a = t[o];
    switch (o) {
      case "inline":
      case "slice":
        typeof a == "boolean" && (r[o] = a);
        break;
      case "hFlip":
      case "vFlip":
        a === !0 && (r[o] = !r[o]);
        break;
      case "hAlign":
      case "vAlign":
        typeof a == "string" && a !== "" && (r[o] = a);
        break;
      case "width":
      case "height":
        (typeof a == "string" && a !== "" || typeof a == "number" && a || a === null) && (r[o] = a);
        break;
      case "rotate":
        typeof a == "number" && (r[o] += a);
        break;
    }
  }
  return r;
}
const Wt = /(-?[0-9.]*[0-9]+[0-9.]*)/g, Zt = /^-?[0-9.]*[0-9]+[0-9.]*$/g;
function ye(e, t, r) {
  if (t === 1)
    return e;
  if (r = r === void 0 ? 100 : r, typeof e == "number")
    return Math.ceil(e * t * r) / r;
  if (typeof e != "string")
    return e;
  const n = e.split(Wt);
  if (n === null || !n.length)
    return e;
  const o = [];
  let a = n.shift(), l = Zt.test(a);
  for (; ; ) {
    if (l) {
      const i = parseFloat(a);
      isNaN(i) ? o.push(a) : o.push(Math.ceil(i * t * r) / r);
    } else
      o.push(a);
    if (a = n.shift(), a === void 0)
      return o.join("");
    l = !l;
  }
}
function er(e) {
  let t = "";
  switch (e.hAlign) {
    case "left":
      t += "xMin";
      break;
    case "right":
      t += "xMax";
      break;
    default:
      t += "xMid";
  }
  switch (e.vAlign) {
    case "top":
      t += "YMin";
      break;
    case "bottom":
      t += "YMax";
      break;
    default:
      t += "YMid";
  }
  return t += e.slice ? " slice" : " meet", t;
}
function tr(e, t) {
  const r = {
    left: e.left,
    top: e.top,
    width: e.width,
    height: e.height
  };
  let n = e.body;
  [e, t].forEach((i) => {
    const c = [], u = i.hFlip, d = i.vFlip;
    let f = i.rotate;
    u ? d ? f += 2 : (c.push("translate(" + (r.width + r.left).toString() + " " + (0 - r.top).toString() + ")"), c.push("scale(-1 1)"), r.top = r.left = 0) : d && (c.push("translate(" + (0 - r.left).toString() + " " + (r.height + r.top).toString() + ")"), c.push("scale(1 -1)"), r.top = r.left = 0);
    let p;
    switch (f < 0 && (f -= Math.floor(f / 4) * 4), f = f % 4, f) {
      case 1:
        p = r.height / 2 + r.top, c.unshift("rotate(90 " + p.toString() + " " + p.toString() + ")");
        break;
      case 2:
        c.unshift("rotate(180 " + (r.width / 2 + r.left).toString() + " " + (r.height / 2 + r.top).toString() + ")");
        break;
      case 3:
        p = r.width / 2 + r.left, c.unshift("rotate(-90 " + p.toString() + " " + p.toString() + ")");
        break;
    }
    f % 2 === 1 && ((r.left !== 0 || r.top !== 0) && (p = r.left, r.left = r.top, r.top = p), r.width !== r.height && (p = r.width, r.width = r.height, r.height = p)), c.length && (n = '<g transform="' + c.join(" ") + '">' + n + "</g>");
  });
  let o, a;
  t.width === null && t.height === null ? (a = "1em", o = ye(a, r.width / r.height)) : t.width !== null && t.height !== null ? (o = t.width, a = t.height) : t.height !== null ? (a = t.height, o = ye(a, r.width / r.height)) : (o = t.width, a = ye(o, r.height / r.width)), o === "auto" && (o = r.width), a === "auto" && (a = r.height), o = typeof o == "string" ? o : o.toString() + "", a = typeof a == "string" ? a : a.toString() + "";
  const l = {
    attributes: {
      width: o,
      height: a,
      preserveAspectRatio: er(t),
      viewBox: r.left.toString() + " " + r.top.toString() + " " + r.width.toString() + " " + r.height.toString()
    },
    body: n
  };
  return t.inline && (l.inline = !0), l;
}
const rr = /\sid="(\S+)"/g, or = "IconifyId" + Date.now().toString(16) + (Math.random() * 16777216 | 0).toString(16);
let nr = 0;
function ar(e, t = or) {
  const r = [];
  let n;
  for (; n = rr.exec(e); )
    r.push(n[1]);
  return r.length && r.forEach((o) => {
    const a = typeof t == "function" ? t(o) : t + (nr++).toString(), l = o.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
    e = e.replace(new RegExp('([#;"])(' + l + ')([")]|\\.[a-z])', "g"), "$1" + a + "$3");
  }), e;
}
const Ce = /* @__PURE__ */ Object.create(null);
function lr(e, t) {
  Ce[e] = t;
}
function Se(e) {
  return Ce[e] || Ce[""];
}
function Le(e) {
  let t;
  if (typeof e.resources == "string")
    t = [e.resources];
  else if (t = e.resources, !(t instanceof Array) || !t.length)
    return null;
  return {
    resources: t,
    path: e.path === void 0 ? "/" : e.path,
    maxURL: e.maxURL ? e.maxURL : 500,
    rotate: e.rotate ? e.rotate : 750,
    timeout: e.timeout ? e.timeout : 5e3,
    random: e.random === !0,
    index: e.index ? e.index : 0,
    dataAfterTimeout: e.dataAfterTimeout !== !1
  };
}
const Fe = /* @__PURE__ */ Object.create(null), re = [
  "https://api.simplesvg.com",
  "https://api.unisvg.com"
], ue = [];
for (; re.length > 0; )
  re.length === 1 || Math.random() > 0.5 ? ue.push(re.shift()) : ue.push(re.pop());
Fe[""] = Le({
  resources: ["https://api.iconify.design"].concat(ue)
});
function sr(e, t) {
  const r = Le(t);
  return r === null ? !1 : (Fe[e] = r, !0);
}
function Re(e) {
  return Fe[e];
}
const Xe = (e, t) => {
  let r = e, n = r.indexOf("?") !== -1;
  function o(a) {
    switch (typeof a) {
      case "boolean":
        return a ? "true" : "false";
      case "number":
        return encodeURIComponent(a);
      case "string":
        return encodeURIComponent(a);
      default:
        throw new Error("Invalid parameter");
    }
  }
  return Object.keys(t).forEach((a) => {
    let l;
    try {
      l = o(t[a]);
    } catch {
      return;
    }
    r += (n ? "&" : "?") + encodeURIComponent(a) + "=" + l, n = !0;
  }), r;
}, We = {}, de = {}, ir = () => {
  let e;
  try {
    if (e = fetch, typeof e == "function")
      return e;
  } catch {
  }
  return null;
};
let _e = ir();
function cr(e, t) {
  const r = Re(e);
  if (!r)
    return 0;
  let n;
  if (!r.maxURL)
    n = 0;
  else {
    let a = 0;
    r.resources.forEach((i) => {
      a = Math.max(a, i.length);
    });
    const l = Xe(t + ".json", {
      icons: ""
    });
    n = r.maxURL - a - r.path.length - l.length;
  }
  const o = e + ":" + t;
  return de[e] = r.path, We[o] = n, n;
}
function ur(e) {
  return e === 404;
}
const dr = (e, t, r) => {
  const n = [];
  let o = We[t];
  o === void 0 && (o = cr(e, t));
  const a = "icons";
  let l = {
    type: a,
    provider: e,
    prefix: t,
    icons: []
  }, i = 0;
  return r.forEach((c, u) => {
    i += c.length + 1, i >= o && u > 0 && (n.push(l), l = {
      type: a,
      provider: e,
      prefix: t,
      icons: []
    }, i = c.length), l.icons.push(c);
  }), n.push(l), n;
};
function fr(e) {
  if (typeof e == "string") {
    if (de[e] === void 0) {
      const t = Re(e);
      if (!t)
        return "/";
      de[e] = t.path;
    }
    return de[e];
  }
  return "/";
}
const pr = (e, t, r) => {
  if (!_e) {
    r("abort", 424);
    return;
  }
  let n = fr(t.provider);
  switch (t.type) {
    case "icons": {
      const a = t.prefix, i = t.icons.join(",");
      n += Xe(a + ".json", {
        icons: i
      });
      break;
    }
    case "custom": {
      const a = t.uri;
      n += a.slice(0, 1) === "/" ? a.slice(1) : a;
      break;
    }
    default:
      r("abort", 400);
      return;
  }
  let o = 503;
  _e(e + n).then((a) => {
    const l = a.status;
    if (l !== 200) {
      setTimeout(() => {
        r(ur(l) ? "abort" : "next", l);
      });
      return;
    }
    return o = 501, a.json();
  }).then((a) => {
    if (typeof a != "object" || a === null) {
      setTimeout(() => {
        r("next", o);
      });
      return;
    }
    setTimeout(() => {
      r("success", a);
    });
  }).catch(() => {
    r("next", o);
  });
}, gr = {
  prepare: dr,
  send: pr
};
function hr(e) {
  const t = {
    loaded: [],
    missing: [],
    pending: []
  }, r = /* @__PURE__ */ Object.create(null);
  e.sort((o, a) => o.provider !== a.provider ? o.provider.localeCompare(a.provider) : o.prefix !== a.prefix ? o.prefix.localeCompare(a.prefix) : o.name.localeCompare(a.name));
  let n = {
    provider: "",
    prefix: "",
    name: ""
  };
  return e.forEach((o) => {
    if (n.name === o.name && n.prefix === o.prefix && n.provider === o.provider)
      return;
    n = o;
    const a = o.provider, l = o.prefix, i = o.name;
    r[a] === void 0 && (r[a] = /* @__PURE__ */ Object.create(null));
    const c = r[a];
    c[l] === void 0 && (c[l] = Z(a, l));
    const u = c[l];
    let d;
    u.icons[i] !== void 0 ? d = t.loaded : l === "" || u.missing[i] !== void 0 ? d = t.missing : d = t.pending;
    const f = {
      provider: a,
      prefix: l,
      name: i
    };
    d.push(f);
  }), t;
}
const Y = /* @__PURE__ */ Object.create(null), be = /* @__PURE__ */ Object.create(null);
function Ze(e, t) {
  e.forEach((r) => {
    const n = r.provider;
    if (Y[n] === void 0)
      return;
    const o = Y[n], a = r.prefix, l = o[a];
    l && (o[a] = l.filter((i) => i.id !== t));
  });
}
function yr(e, t) {
  be[e] === void 0 && (be[e] = /* @__PURE__ */ Object.create(null));
  const r = be[e];
  r[t] || (r[t] = !0, setTimeout(() => {
    if (r[t] = !1, Y[e] === void 0 || Y[e][t] === void 0)
      return;
    const n = Y[e][t].slice(0);
    if (!n.length)
      return;
    const o = Z(e, t);
    let a = !1;
    n.forEach((l) => {
      const i = l.icons, c = i.pending.length;
      i.pending = i.pending.filter((u) => {
        if (u.prefix !== t)
          return !0;
        const d = u.name;
        if (o.icons[d] !== void 0)
          i.loaded.push({
            provider: e,
            prefix: t,
            name: d
          });
        else if (o.missing[d] !== void 0)
          i.missing.push({
            provider: e,
            prefix: t,
            name: d
          });
        else
          return a = !0, !0;
        return !1;
      }), i.pending.length !== c && (a || Ze([
        {
          provider: e,
          prefix: t
        }
      ], l.id), l.callback(i.loaded.slice(0), i.missing.slice(0), i.pending.slice(0), l.abort));
    });
  }));
}
let br = 0;
function mr(e, t, r) {
  const n = br++, o = Ze.bind(null, r, n);
  if (!t.pending.length)
    return o;
  const a = {
    id: n,
    icons: t,
    callback: e,
    abort: o
  };
  return r.forEach((l) => {
    const i = l.provider, c = l.prefix;
    Y[i] === void 0 && (Y[i] = /* @__PURE__ */ Object.create(null));
    const u = Y[i];
    u[c] === void 0 && (u[c] = []), u[c].push(a);
  }), o;
}
function vr(e, t = !0, r = !1) {
  const n = [];
  return e.forEach((o) => {
    const a = typeof o == "string" ? ge(o, !1, r) : o;
    (!t || ne(a, r)) && n.push({
      provider: a.provider,
      prefix: a.prefix,
      name: a.name
    });
  }), n;
}
var Me = {
  resources: [],
  index: 0,
  timeout: 2e3,
  rotate: 750,
  random: !1,
  dataAfterTimeout: !1
};
function xr(e, t, r, n) {
  const o = e.resources.length, a = e.random ? Math.floor(Math.random() * o) : e.index;
  let l;
  if (e.random) {
    let k = e.resources.slice(0);
    for (l = []; k.length > 1; ) {
      const F = Math.floor(Math.random() * k.length);
      l.push(k[F]), k = k.slice(0, F).concat(k.slice(F + 1));
    }
    l = l.concat(k);
  } else
    l = e.resources.slice(a).concat(e.resources.slice(0, a));
  const i = Date.now();
  let c = "pending", u = 0, d, f = null, p = [], g = [];
  typeof n == "function" && g.push(n);
  function T() {
    f && (clearTimeout(f), f = null);
  }
  function y() {
    c === "pending" && (c = "aborted"), T(), p.forEach((k) => {
      k.status === "pending" && (k.status = "aborted");
    }), p = [];
  }
  function j(k, F) {
    F && (g = []), typeof k == "function" && g.push(k);
  }
  function x() {
    return {
      startTime: i,
      payload: t,
      status: c,
      queriesSent: u,
      queriesPending: p.length,
      subscribe: j,
      abort: y
    };
  }
  function v() {
    c = "failed", g.forEach((k) => {
      k(void 0, d);
    });
  }
  function U() {
    p.forEach((k) => {
      k.status === "pending" && (k.status = "aborted");
    }), p = [];
  }
  function ee(k, F, H) {
    const _ = F !== "success";
    switch (p = p.filter((b) => b !== k), c) {
      case "pending":
        break;
      case "failed":
        if (_ || !e.dataAfterTimeout)
          return;
        break;
      default:
        return;
    }
    if (F === "abort") {
      d = H, v();
      return;
    }
    if (_) {
      d = H, p.length || (l.length ? D() : v());
      return;
    }
    if (T(), U(), !e.random) {
      const b = e.resources.indexOf(k.resource);
      b !== -1 && b !== e.index && (e.index = b);
    }
    c = "completed", g.forEach((b) => {
      b(H);
    });
  }
  function D() {
    if (c !== "pending")
      return;
    T();
    const k = l.shift();
    if (k === void 0) {
      if (p.length) {
        f = setTimeout(() => {
          T(), c === "pending" && (U(), v());
        }, e.timeout);
        return;
      }
      v();
      return;
    }
    const F = {
      status: "pending",
      resource: k,
      callback: (H, _) => {
        ee(F, H, _);
      }
    };
    p.push(F), u++, f = setTimeout(D, e.rotate), r(k, t, F.callback);
  }
  return setTimeout(D), x;
}
function wr(e) {
  if (typeof e != "object" || typeof e.resources != "object" || !(e.resources instanceof Array) || !e.resources.length)
    throw new Error("Invalid Reduncancy configuration");
  const t = /* @__PURE__ */ Object.create(null);
  let r;
  for (r in Me)
    e[r] !== void 0 ? t[r] = e[r] : t[r] = Me[r];
  return t;
}
function et(e) {
  const t = wr(e);
  let r = [];
  function n() {
    r = r.filter((i) => i().status === "pending");
  }
  function o(i, c, u) {
    const d = xr(t, i, c, (f, p) => {
      n(), u && u(f, p);
    });
    return r.push(d), d;
  }
  function a(i) {
    const c = r.find((u) => i(u));
    return c !== void 0 ? c : null;
  }
  return {
    query: o,
    find: a,
    setIndex: (i) => {
      t.index = i;
    },
    getIndex: () => t.index,
    cleanup: n
  };
}
function Ne() {
}
const me = /* @__PURE__ */ Object.create(null);
function kr(e) {
  if (me[e] === void 0) {
    const t = Re(e);
    if (!t)
      return;
    const r = et(t), n = {
      config: t,
      redundancy: r
    };
    me[e] = n;
  }
  return me[e];
}
function Cr(e, t, r) {
  let n, o;
  if (typeof e == "string") {
    const a = Se(e);
    if (!a)
      return r(void 0, 424), Ne;
    o = a.send;
    const l = kr(e);
    l && (n = l.redundancy);
  } else {
    const a = Le(e);
    if (a) {
      n = et(a);
      const l = e.resources ? e.resources[0] : "", i = Se(l);
      i && (o = i.send);
    }
  }
  return !n || !o ? (r(void 0, 424), Ne) : n.query(t, o, r)().abort;
}
const Te = {};
function Ue() {
}
const X = /* @__PURE__ */ Object.create(null), ve = /* @__PURE__ */ Object.create(null), xe = /* @__PURE__ */ Object.create(null), we = /* @__PURE__ */ Object.create(null);
function Sr(e, t) {
  xe[e] === void 0 && (xe[e] = /* @__PURE__ */ Object.create(null));
  const r = xe[e];
  r[t] || (r[t] = !0, setTimeout(() => {
    r[t] = !1, yr(e, t);
  }));
}
const De = /* @__PURE__ */ Object.create(null);
function Tr(e, t, r) {
  function n() {
    const i = (e === "" ? "" : "@" + e + ":") + t, c = Math.floor(Date.now() / 6e4);
    De[i] < c && (De[i] = c, console.error('Unable to retrieve icons for "' + i + '" because API is not configured properly.'));
  }
  ve[e] === void 0 && (ve[e] = /* @__PURE__ */ Object.create(null));
  const o = ve[e];
  we[e] === void 0 && (we[e] = /* @__PURE__ */ Object.create(null));
  const a = we[e];
  X[e] === void 0 && (X[e] = /* @__PURE__ */ Object.create(null));
  const l = X[e];
  o[t] === void 0 ? o[t] = r : o[t] = o[t].concat(r).sort(), a[t] || (a[t] = !0, setTimeout(() => {
    a[t] = !1;
    const i = o[t];
    delete o[t];
    const c = Se(e);
    if (!c) {
      n();
      return;
    }
    c.prepare(e, t, i).forEach((d) => {
      Cr(e, d, (f, p) => {
        const g = Z(e, t);
        if (typeof f != "object") {
          if (p !== 404)
            return;
          const T = Date.now();
          d.icons.forEach((y) => {
            g.missing[y] = T;
          });
        } else
          try {
            const T = je(g, f);
            if (!T.length)
              return;
            const y = l[t];
            T.forEach((j) => {
              delete y[j];
            }), Te.store && Te.store(e, f);
          } catch (T) {
            console.error(T);
          }
        Sr(e, t);
      });
    });
  }));
}
const Ir = (e, t) => {
  const r = vr(e, !0, Je()), n = hr(r);
  if (!n.pending.length) {
    let u = !0;
    return t && setTimeout(() => {
      u && t(n.loaded, n.missing, n.pending, Ue);
    }), () => {
      u = !1;
    };
  }
  const o = /* @__PURE__ */ Object.create(null), a = [];
  let l, i;
  n.pending.forEach((u) => {
    const d = u.provider, f = u.prefix;
    if (f === i && d === l)
      return;
    l = d, i = f, a.push({
      provider: d,
      prefix: f
    }), X[d] === void 0 && (X[d] = /* @__PURE__ */ Object.create(null));
    const p = X[d];
    p[f] === void 0 && (p[f] = /* @__PURE__ */ Object.create(null)), o[d] === void 0 && (o[d] = /* @__PURE__ */ Object.create(null));
    const g = o[d];
    g[f] === void 0 && (g[f] = []);
  });
  const c = Date.now();
  return n.pending.forEach((u) => {
    const d = u.provider, f = u.prefix, p = u.name, g = X[d][f];
    g[p] === void 0 && (g[p] = c, o[d][f].push(p));
  }), a.forEach((u) => {
    const d = u.provider, f = u.prefix;
    o[d][f].length && Tr(d, f, o[d][f]);
  }), t ? mr(t, n, a) : Ue;
}, tt = "iconify2", le = "iconify", rt = le + "-count", ot = le + "-version", nt = 36e5, Br = 168, Ve = {
  local: !0,
  session: !0
};
let Ie = !1;
const at = {
  local: 0,
  session: 0
}, lt = {
  local: [],
  session: []
};
let se = typeof window > "u" ? {} : window;
function st(e) {
  const t = e + "Storage";
  try {
    if (se && se[t] && typeof se[t].length == "number")
      return se[t];
  } catch {
  }
  return Ve[e] = !1, null;
}
function Oe(e, t, r) {
  try {
    return e.setItem(rt, r.toString()), at[t] = r, !0;
  } catch {
    return !1;
  }
}
function it(e) {
  const t = e.getItem(rt);
  if (t) {
    const r = parseInt(t);
    return r || 0;
  }
  return 0;
}
function jr(e, t) {
  try {
    e.setItem(ot, tt);
  } catch {
  }
  Oe(e, t, 0);
}
function Lr(e) {
  try {
    const t = it(e);
    for (let r = 0; r < t; r++)
      e.removeItem(le + r.toString());
  } catch {
  }
}
const ct = () => {
  if (Ie)
    return;
  Ie = !0;
  const e = Math.floor(Date.now() / nt) - Br;
  function t(r) {
    const n = st(r);
    if (!n)
      return;
    const o = (a) => {
      const l = le + a.toString(), i = n.getItem(l);
      if (typeof i != "string")
        return !1;
      let c = !0;
      try {
        const u = JSON.parse(i);
        if (typeof u != "object" || typeof u.cached != "number" || u.cached < e || typeof u.provider != "string" || typeof u.data != "object" || typeof u.data.prefix != "string")
          c = !1;
        else {
          const d = u.provider, f = u.data.prefix, p = Z(d, f);
          c = je(p, u.data).length > 0;
        }
      } catch {
        c = !1;
      }
      return c || n.removeItem(l), c;
    };
    try {
      const a = n.getItem(ot);
      if (a !== tt) {
        a && Lr(n), jr(n, r);
        return;
      }
      let l = it(n);
      for (let i = l - 1; i >= 0; i--)
        o(i) || (i === l - 1 ? l-- : lt[r].push(i));
      Oe(n, r, l);
    } catch {
    }
  }
  for (const r in Ve)
    t(r);
}, Fr = (e, t) => {
  Ie || ct();
  function r(n) {
    if (!Ve[n])
      return !1;
    const o = st(n);
    if (!o)
      return !1;
    let a = lt[n].shift();
    if (a === void 0 && (a = at[n], !Oe(o, n, a + 1)))
      return !1;
    try {
      const l = {
        cached: Math.floor(Date.now() / nt),
        provider: e,
        data: t
      };
      o.setItem(le + a.toString(), JSON.stringify(l));
    } catch {
      return !1;
    }
    return !0;
  }
  !Object.keys(t.icons).length || (t.not_found && (t = Object.assign({}, t), delete t.not_found), r("local") || r("session"));
}, ut = /[\s,]+/;
function Rr(e, t) {
  t.split(ut).forEach((r) => {
    switch (r.trim()) {
      case "horizontal":
        e.hFlip = !0;
        break;
      case "vertical":
        e.vFlip = !0;
        break;
    }
  });
}
function Vr(e, t) {
  t.split(ut).forEach((r) => {
    const n = r.trim();
    switch (n) {
      case "left":
      case "center":
      case "right":
        e.hAlign = n;
        break;
      case "top":
      case "middle":
      case "bottom":
        e.vAlign = n;
        break;
      case "slice":
      case "crop":
        e.slice = !0;
        break;
      case "meet":
        e.slice = !1;
    }
  });
}
function Or(e, t = 0) {
  const r = e.replace(/^-?[0-9.]*/, "");
  function n(o) {
    for (; o < 0; )
      o += 4;
    return o % 4;
  }
  if (r === "") {
    const o = parseInt(e);
    return isNaN(o) ? 0 : n(o);
  } else if (r !== e) {
    let o = 0;
    switch (r) {
      case "%":
        o = 25;
        break;
      case "deg":
        o = 90;
    }
    if (o) {
      let a = parseFloat(e.slice(0, e.length - r.length));
      return isNaN(a) ? 0 : (a = a / o, a % 1 === 0 ? n(a) : 0);
    }
  }
  return t;
}
const Ar = {
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  "aria-hidden": !0,
  role: "img"
};
let K = {};
["horizontal", "vertical"].forEach((e) => {
  ["Align", "Flip"].forEach((t) => {
    const n = {
      attr: e.slice(0, 1) + t,
      boolean: t === "Flip"
    };
    K[e + "-" + t.toLowerCase()] = n, K[e.slice(0, 1) + "-" + t.toLowerCase()] = n, K[e + t] = n;
  });
});
const $e = (e, t) => {
  const r = Xt(Pe, t), n = { ...Ar };
  let o = typeof t.style == "object" && !(t.style instanceof Array) ? { ...t.style } : {};
  for (let c in t) {
    const u = t[c];
    if (u !== void 0)
      switch (c) {
        case "icon":
        case "style":
        case "onLoad":
          break;
        case "inline":
        case "hFlip":
        case "vFlip":
          r[c] = u === !0 || u === "true" || u === 1;
          break;
        case "flip":
          typeof u == "string" && Rr(r, u);
          break;
        case "align":
          typeof u == "string" && Vr(r, u);
          break;
        case "color":
          o.color = u;
          break;
        case "rotate":
          typeof u == "string" ? r[c] = Or(u) : typeof u == "number" && (r[c] = u);
          break;
        case "ariaHidden":
        case "aria-hidden":
          u !== !0 && u !== "true" && delete n["aria-hidden"];
          break;
        default:
          K[c] !== void 0 ? K[c].boolean && (u === !0 || u === "true" || u === 1) ? r[K[c].attr] = !0 : !K[c].boolean && typeof u == "string" && u !== "" && (r[K[c].attr] = u) : Pe[c] === void 0 && (n[c] = u);
      }
  }
  const a = tr(e, r);
  for (let c in a.attributes)
    n[c] = a.attributes[c];
  a.inline && o.verticalAlign === void 0 && o["vertical-align"] === void 0 && (o.verticalAlign = "-0.125em");
  let l = 0, i = t.id;
  return typeof i == "string" && (i = i.replace(/-/g, "_")), n.innerHTML = ar(a.body, i ? () => i + "ID" + l++ : "iconifyVue"), Object.keys(o).length > 0 && (n.style = o), ht("svg", n);
};
Je(!0);
lr("", gr);
if (typeof document < "u" && typeof window < "u") {
  Te.store = Fr, ct();
  const e = window;
  if (e.IconifyPreload !== void 0) {
    const t = e.IconifyPreload, r = "Invalid IconifyPreload syntax.";
    typeof t == "object" && t !== null && (t instanceof Array ? t : [t]).forEach((n) => {
      try {
        (typeof n != "object" || n === null || n instanceof Array || typeof n.icons != "object" || typeof n.prefix != "string" || !Jt(n)) && console.error(r);
      } catch {
        console.error(r);
      }
    });
  }
  if (e.IconifyProviders !== void 0) {
    const t = e.IconifyProviders;
    if (typeof t == "object" && t !== null)
      for (let r in t) {
        const n = "IconifyProviders[" + r + "] is invalid.";
        try {
          const o = t[r];
          if (typeof o != "object" || !o || o.resources === void 0)
            continue;
          sr(r, o) || console.error(n);
        } catch {
          console.error(n);
        }
      }
  }
}
const Er = pe({
  body: ""
}), ze = N({
  inheritAttrs: !1,
  data() {
    return {
      iconMounted: !1,
      counter: 0
    };
  },
  mounted() {
    this._name = "", this._loadingIcon = null, this.iconMounted = !0;
  },
  unmounted() {
    this.abortLoading();
  },
  methods: {
    abortLoading() {
      this._loadingIcon && (this._loadingIcon.abort(), this._loadingIcon = null);
    },
    getIcon(e, t) {
      if (typeof e == "object" && e !== null && typeof e.body == "string")
        return this._name = "", this.abortLoading(), {
          data: pe(e)
        };
      let r;
      if (typeof e != "string" || (r = ge(e, !1, !0)) === null)
        return this.abortLoading(), null;
      const n = Yt(r);
      if (n === null)
        return (!this._loadingIcon || this._loadingIcon.name !== e) && (this.abortLoading(), this._name = "", this._loadingIcon = {
          name: e,
          abort: Ir([r], () => {
            this.counter++;
          })
        }), null;
      this.abortLoading(), this._name !== e && (this._name = e, t && t(e));
      const o = ["iconify"];
      return r.prefix !== "" && o.push("iconify--" + r.prefix), r.provider !== "" && o.push("iconify--" + r.provider), { data: n, classes: o };
    }
  },
  render() {
    this.counter;
    const e = this.$attrs, t = this.iconMounted ? this.getIcon(e.icon, e.onLoad) : null;
    if (!t)
      return $e(Er, e);
    let r = e;
    return t.classes && (r = {
      ...e,
      class: (typeof e.class == "string" ? e.class + " " : "") + t.classes.join(" ")
    }), $e(t.data, r);
  }
}), Pr = { class: "flex gap-2" }, _r = /* @__PURE__ */ N({
  __name: "VFileUploadActions",
  props: {
    readonly: { type: Boolean },
    button: { type: Boolean },
    full: { type: Boolean },
    disabled: { type: Boolean },
    hideRemove: { type: Boolean },
    changeText: { default: "Change" },
    removeText: { default: "Remove" }
  },
  emits: ["choose", "remove"],
  setup(e, { emit: t }) {
    return (r, n) => (h(), w("div", {
      class: I(["flex w-full mt-3 justify-center items-center gap-y-2 gap-x-2", [e.full || e.button ? "flex-row" : "w-full sm:w-[180px] flex-col"]])
    }, [
      m("div", null, [
        R(r.$slots, "prepend")
      ]),
      m("div", Pr, [
        O(s(W), {
          disabled: e.readonly || e.disabled,
          small: "",
          dense: "",
          type: "button",
          onClick: n[0] || (n[0] = (o) => t("choose"))
        }, {
          default: A(() => [
            q(C(e.changeText), 1)
          ]),
          _: 1
        }, 8, ["disabled"]),
        e.hideRemove ? B("", !0) : (h(), P(s(W), {
          key: 0,
          small: "",
          dense: "",
          type: "button",
          color: "error",
          disabled: e.readonly || e.disabled,
          onClick: n[1] || (n[1] = (o) => t("remove"))
        }, {
          default: A(() => [
            q(C(e.removeText), 1)
          ]),
          _: 1
        }, 8, ["disabled"]))
      ]),
      m("div", null, [
        R(r.$slots, "append")
      ])
    ], 2));
  }
}), Mr = ["title"], Nr = { class: "space-x-1 px-1" }, Ur = /* @__PURE__ */ N({
  __name: "VFileUploadDefaultTheme",
  props: {
    borderClass: { default: "" },
    fileName: null,
    placeholder: null,
    readonly: { type: Boolean },
    disabled: { type: Boolean },
    hasFile: { type: Boolean },
    changeText: { default: "Change" },
    browseText: null,
    hideRemove: { type: Boolean },
    removeText: { default: "Remove" }
  },
  emits: ["remove", "choose"],
  setup(e, { emit: t }) {
    return (r, n) => (h(), w("div", {
      class: I(["transition duration-300 border group rounded-lg flex gap-4 items-center py-1", e.borderClass])
    }, [
      m("div", {
        class: "flex-1 text-gray-500 hover:text-gray-700 text-sm truncate px-3 py-1 h-full",
        title: e.fileName || e.placeholder,
        onClick: n[0] || (n[0] = (o) => t("choose"))
      }, C(e.fileName || e.placeholder), 9, Mr),
      m("div", Nr, [
        O(s(W), {
          disabled: e.readonly || e.disabled,
          type: "button",
          size: "sm",
          onClick: n[1] || (n[1] = (o) => t("choose"))
        }, {
          default: A(() => [
            q(C(e.hasFile ? e.changeText : e.browseText), 1)
          ]),
          _: 1
        }, 8, ["disabled"]),
        e.hasFile && !e.hideRemove ? (h(), P(s(W), {
          key: 0,
          type: "button",
          size: "sm",
          color: "error",
          disabled: e.readonly || e.disabled,
          onClick: n[2] || (n[2] = (o) => t("remove"))
        }, {
          default: A(() => [
            q(C(e.removeText), 1)
          ]),
          _: 1
        }, 8, ["disabled"])) : B("", !0)
      ])
    ], 2));
  }
}), Dr = { class: "flex gap-2" }, $r = {
  key: 0,
  class: "flex gap-2 items-center"
}, zr = {
  key: 1,
  class: "text-sm mt-1 text-gray-500"
}, qr = { class: "flex gap-2 items-center" }, Hr = /* @__PURE__ */ N({
  __name: "VFileUploadButtonTheme",
  props: {
    hasFile: { type: Boolean, default: !1 },
    disabledClass: null,
    borderClass: null,
    fileName: null,
    fileURL: null,
    viewFileText: { default: "View file" },
    readonly: { type: Boolean },
    placeholder: { default: "No file selected" },
    hidePlaceholder: { type: Boolean },
    changeText: { default: "Change" },
    removeText: { default: "Remove" },
    hideRemove: { type: Boolean },
    disabled: { type: Boolean },
    browseText: { default: "Browse" }
  },
  emits: ["choose", "remove"],
  setup(e, { emit: t }) {
    return (r, n) => (h(), w("div", Dr, [
      e.hasFile ? (h(), w("div", $r, [
        m("div", {
          class: I(["px-4 py-2 truncate text-center border rounded hover:border-gray-400", [e.disabledClass, e.borderClass]]),
          onClick: n[0] || (n[0] = (o) => t("choose"))
        }, C(e.fileName), 3),
        e.readonly ? (h(), P(s(W), {
          key: 0,
          href: e.fileURL,
          target: "_blank",
          tag: "a",
          class: "whitespace-nowrap",
          small: "",
          color: "primary"
        }, {
          default: A(() => [
            q(C(e.viewFileText), 1)
          ]),
          _: 1
        }, 8, ["href"])) : B("", !0),
        !e.hasFile && !e.hidePlaceholder ? (h(), w("div", zr, C(e.placeholder), 1)) : B("", !0)
      ])) : B("", !0),
      m("div", qr, [
        O(s(W), {
          uppercase: "",
          outlined: "",
          type: "button",
          color: "primary",
          class: "line-clamp",
          disabled: e.readonly || e.disabled,
          onClick: n[1] || (n[1] = (o) => t("choose"))
        }, {
          default: A(() => [
            R(r.$slots, "icon.plus", {}, () => [
              O(s(Lt), { class: "w-5 h-5 mr-2" })
            ]),
            q(" " + C(e.hasFile ? e.changeText : e.browseText), 1)
          ]),
          _: 3
        }, 8, ["disabled"]),
        e.hasFile && !e.hideRemove ? (h(), P(s(W), {
          key: 0,
          small: "",
          dense: "",
          outlined: "",
          type: "button",
          color: "error",
          class: "space-x-2",
          disabled: e.readonly || e.disabled,
          onClick: n[2] || (n[2] = (o) => t("remove"))
        }, {
          default: A(() => [
            R(r.$slots, "icon.trash", {}, () => [
              O(s(Ft), { class: "w-5 h-5" })
            ]),
            m("span", null, C(e.removeText), 1)
          ]),
          _: 3
        }, 8, ["disabled"])) : B("", !0)
      ])
    ]));
  }
}), Qr = {
  key: 1,
  class: "px-2 text-center"
}, Kr = { class: "uppercase text-gray-500" }, Yr = /* @__PURE__ */ N({
  __name: "VFileUploadImageTheme",
  props: {
    rounded: { type: Boolean, default: !1 },
    sizeClass: null,
    borderClass: null,
    previewURL: null,
    loading: { type: Boolean },
    image: { type: Boolean },
    fileName: null,
    hasFile: { type: Boolean },
    loadingText: { default: "Loading..." },
    browseText: { default: "Browse" }
  },
  emits: ["choose"],
  setup(e, { emit: t }) {
    const r = e, n = S(
      () => r.image && !r.loading ? `url(${r.previewURL})` : "none"
    );
    return (o, a) => (h(), w("div", null, [
      m("button", {
        class: I(["appearance-none transition duration-300 border cursor-pointer flex flex-col items-center justify-center py-10 bg-no-repeat bg-contain bg-center max-w-full", [e.sizeClass, { "rounded-lg": e.rounded }, e.borderClass]]),
        style: qe({ backgroundImage: s(n) }),
        onClick: a[0] || (a[0] = (l) => t("choose"))
      }, [
        e.loading ? (h(), P(s(Ke), {
          key: 0,
          color: "primary",
          large: ""
        })) : e.hasFile ? (h(), w("div", Qr, C(e.image ? "" : e.fileName), 1)) : (h(), w(He, { key: 2 }, [
          O(s(Rt), { class: "w-10 h-10 text-gray-500 mb-1" }),
          m("div", Kr, C(e.loading ? e.loadingText : e.browseText), 1)
        ], 64))
      ], 6)
    ]));
  }
}), Gr = {
  key: 0,
  class: "text-center flex flex-col gap-4"
}, Jr = { class: "flex flex-col gap-4" }, Xr = {
  key: 1,
  class: "text-gray-500 text-sm"
}, Wr = {
  key: 0,
  class: "space-x-3"
}, Zr = {
  key: 1,
  class: "space-y-1 text-center"
}, eo = /* @__PURE__ */ m("svg", {
  class: "mx-auto h-12 w-12 text-gray-400",
  stroke: "currentColor",
  fill: "none",
  viewBox: "0 0 48 48",
  "aria-hidden": "true"
}, [
  /* @__PURE__ */ m("path", {
    d: "M28 8H12a4 4 0 00-4 4v20m32-12v8m0 0v8a4 4 0 01-4 4H12a4 4 0 01-4-4v-4m32-4l-3.172-3.172a4 4 0 00-5.656 0L28 28M8 32l9.172-9.172a4 4 0 015.656 0L28 28m0 0l4 4m4-24h8m-4-4v8m-12 4h.02",
    "stroke-width": "2",
    "stroke-linecap": "round",
    "stroke-linejoin": "round"
  })
], -1), to = { class: "flex text-sm text-gray-600" }, ro = { class: "pl-1" }, oo = { class: "text-xs text-gray-500" }, no = /* @__PURE__ */ N({
  __name: "VFileUploadDropzoneTheme",
  props: {
    modelValue: { default: "" },
    readonly: { type: Boolean, default: !1 },
    hasError: { type: Boolean, default: !1 },
    hasFile: { type: Boolean, default: !1 },
    fileName: { default: "" },
    previewURL: { default: "" },
    image: { type: Boolean, default: !1 },
    preview: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    hideRemove: { type: Boolean, default: !1 },
    changeText: { default: "Change" },
    uploadText: { default: "Upload" },
    dragText: { default: "Drag and drop files" },
    removeText: { default: "Remove" },
    hint: { default: "" },
    previewClass: { default: "" },
    loading: { type: Boolean, default: !1 }
  },
  emits: ["choose", "remove", "update:modelValue", "dropped"],
  setup(e, { emit: t }) {
    const r = e, n = S(() => r.readonly ? "border" : r.hasError ? "border-2 border-error-500 border-dashed" : "border-2 border-gray-300 hover:border-gray-400 border-dashed");
    let o = () => crypto.randomUUID();
    const a = S(() => o()), l = "border-gray-300", i = "border-primary-600", c = (g) => Array.from(g.target.classList).includes(a.value), u = (g) => {
      g.preventDefault(), g.target.classList.remove(i), g.target.classList.add(l), c(g) && t("dropped", g.dataTransfer.files);
    }, d = (g) => {
      g.preventDefault();
    }, f = (g) => {
      c(g) && (g.target.classList.remove(l), g.target.classList.add(i));
    }, p = (g) => {
      c(g) && (g.target.classList.remove(i), g.target.classList.add(l));
    };
    return yt(() => {
      document.addEventListener("dragover", d, !1), document.addEventListener("drop", u), document.addEventListener("dragenter", f, !1), document.addEventListener("dragleave", p, !1);
    }), bt(() => {
      document.removeEventListener("dragover", d, !1), document.removeEventListener("drop", u), document.removeEventListener("dragenter", f, !1), document.removeEventListener("dragleave", p, !1);
    }), (g, T) => (h(), w("div", {
      class: I(["flex justify-center items-center p-4 rounded-md transition duration-300", [s(a), s(n)]])
    }, [
      e.hasFile ? (h(), w("div", Gr, [
        R(g.$slots, "dropzone.preview", {
          value: e.modelValue,
          fileName: e.fileName,
          hasFile: e.hasFile
        }, () => [
          m("div", Jr, [
            e.image || e.preview ? (h(), w("div", {
              key: 0,
              class: I(["w-60 h-40 flex bg-contain bg-gray-100 mx-auto rounded-lg bg-no-repeat bg-center", e.previewClass]),
              style: qe({
                backgroundImage: e.loading ? "none" : `url(${e.previewURL})`
              })
            }, null, 6)) : B("", !0),
            e.fileName ? (h(), w("div", Xr, C(e.fileName), 1)) : B("", !0)
          ])
        ]),
        !e.readonly && !e.disabled ? (h(), w("div", Wr, [
          m("button", {
            type: "button",
            class: "appearance-none relative cursor-pointer bg-white rounded-md font-medium text-primary-600 hover:text-primary-500 focus-within:outline-none focus-within:ring-2 focus-within:ring-offset-2 focus-within:ring-primary-500",
            onClick: T[0] || (T[0] = (y) => t("choose"))
          }, [
            m("span", null, C(e.changeText), 1)
          ]),
          e.hideRemove ? B("", !0) : (h(), w("button", {
            key: 0,
            type: "button",
            class: "appearance-none relative cursor-pointer bg-white rounded-md font-medium text-error-600 hover:text-error-500 focus-within:outline-none focus-within:ring-2 focus-within:ring-offset-2 focus-within:ring-error-500",
            onClick: T[1] || (T[1] = (y) => t("remove"))
          }, [
            m("span", null, C(e.removeText), 1)
          ]))
        ])) : B("", !0)
      ])) : (h(), w("div", Zr, [
        R(g.$slots, "dropzone.image", {}, () => [
          eo
        ]),
        m("div", to, [
          m("button", {
            type: "button",
            class: "appearance-none relative cursor-pointer bg-white rounded-md font-medium text-primary-600 hover:text-primary-500 focus-within:outline-none focus-within:ring-2 focus-within:ring-offset-2 focus-within:ring-primary-500",
            onClick: T[2] || (T[2] = (y) => t("choose"))
          }, [
            m("span", null, C(e.uploadText), 1)
          ]),
          m("p", ro, C(e.dragText), 1)
        ]),
        m("p", oo, C(e.hint), 1)
      ]))
    ], 2));
  }
}), ao = ["for"], lo = ["id", "readonly", "accept"], so = {
  key: 0,
  class: "text-xs mt-2 text-black"
}, io = N({
  inheritAttrs: !1
});
({
  ...io
});
const co = ["for"], uo = {
  key: 2,
  class: "relative w-full flex gap-2 items-center"
}, fo = ["id", "placeholder", "type", "disabled", "readonly"], po = {
  key: 3,
  class: "text-error-500 text-sm"
}, go = {
  inheritAttrs: !1
}, ho = /* @__PURE__ */ N({
  ...go,
  __name: "VInput",
  props: {
    value: {
      type: [String, Number],
      default: ""
    },
    modelValue: {
      type: [String, Number],
      default: ""
    },
    type: {
      type: String,
      default: "text"
    },
    name: {
      type: String,
      default: ""
    },
    error: {
      type: Boolean,
      default: !1
    },
    errorMessages: {
      type: Array,
      default: () => []
    },
    readonly: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    size: {
      type: String,
      default: "default"
    },
    placeholder: {
      type: String,
      default: ""
    },
    prependIcon: {
      type: String,
      default: ""
    },
    appendIcon: {
      type: String,
      default: ""
    },
    color: {
      type: String,
      default: "default"
    },
    text: {
      type: Boolean,
      default: !1
    },
    shadow: {
      type: Boolean,
      default: !1
    },
    classes: {
      type: Object,
      default: () => ({
        wrapper: "",
        input: "",
        prepend: "",
        append: ""
      })
    },
    label: {
      type: String,
      default: ""
    },
    rules: {
      type: [Object, String],
      default: null
    },
    id: {
      type: String,
      default: ""
    },
    inputClass: {
      type: String,
      default: ""
    },
    wrapperClass: {
      type: String,
      default: ""
    },
    prependClass: {
      type: String,
      default: ""
    },
    prependIconClass: {
      type: String,
      default: ""
    },
    appendClass: {
      type: String,
      default: ""
    },
    appendIconClass: {
      type: String,
      default: ""
    }
  },
  emits: [
    "input:modelValue",
    "blur",
    "change",
    "clickPrepend",
    "clickPrependIcon",
    "clickAppend",
    "clickAppendIcon"
  ],
  setup(e, { emit: t }) {
    const r = e, { type: n, readonly: o, disabled: a, placeholder: l, prependIcon: i, name: c, rules: u } = fe(r), d = S(() => ({
      xs: "text-xs",
      sm: "text-sm",
      default: "text-base",
      md: "text-base",
      lg: "text-lg",
      xl: "text-xl"
    })[r.size]), { value: f, errorMessage: p } = Be(c, u, {
      initialValue: r.modelValue || r.value
    }), g = S(() => r.error || p.value ? "border-error-500 focus:ring-error-500 focus:ring-opacity-50 focus:border-error-500" : {
      default: "border-gray-300 hover:border-gray-400 focus:ring-primary-500 focus:ring-opacity-50 focus:border-primary-500",
      blue: "border-gray-300 hover:border-gray-400 focus:ring-blue-500 focus:ring-opacity-50 focus:border-blue-500",
      primary: "border-gray-300 hover:border-gray-400 focus:ring-primary-500 focus:ring-opacity-50 focus:border-primary-500",
      secondary: "border-gray-300 hover:border-gray-400 focus:ring-secondary-500 focus:ring-opacity-50 focus:border-secondary-500",
      info: "border-gray-300 hover:border-gray-400 focus:ring-info-500 focus:ring-opacity-50 focus:border-info-500",
      warning: "border-gray-300 hover:border-gray-400 focus:ring-warning-500 focus:ring-opacity-50 focus:border-warning-500",
      success: "border-gray-300 hover:border-gray-400 focus:ring-success-500 focus:ring-opacity-50 focus:border-success-500",
      error: "border-gray-300 hover:border-gray-400 focus:ring-error-500 focus:ring-opacity-50 focus:border-error-500",
      dark: "border-gray-300 hover:border-gray-400 focus:ring-gray-500 focus:ring-opacity-50 focus:border-gray-500"
    }[r.color]), T = S(() => {
      const y = [];
      return r.prependIcon && y.push("pl-10"), r.appendIcon && y.push("pr-10"), y.join(" ");
    });
    return (y, j) => (h(), w("div", {
      class: I(e.wrapperClass)
    }, [
      e.label ? (h(), w("label", {
        key: 0,
        for: e.id || s(c),
        class: "mb-1 block"
      }, C(e.label), 9, co)) : B("", !0),
      e.text ? (h(), w("div", mt($({ key: 1 }, y.$attrs)), C(s(f)), 17)) : (h(), w("div", uo, [
        R(y.$slots, "prepend.outer", {}, () => [
          m("div", {
            class: I(["absolute inset-y-0 left-0 h-full flex items-center px-3 text-gray-500", e.prependClass]),
            onClick: j[1] || (j[1] = (x) => t("clickPrepend"))
          }, [
            R(y.$slots, "prepend", {}, () => [
              O(s(ze), {
                icon: s(i),
                class: I(["w-5 h-5", e.prependIconClass]),
                onClick: j[0] || (j[0] = (x) => t("clickPrependIcon"))
              }, null, 8, ["icon", "class"])
            ])
          ], 2)
        ]),
        vt(m("input", $({
          id: e.id || s(c),
          "onUpdate:modelValue": j[2] || (j[2] = (x) => Qe(f) ? f.value = x : null),
          class: ["w-full border px-3 py-2 focus:outline-none rounded-md transition duration-300 disabled:cursor-not-allowed focus:ring-2", [
            { shadow: e.shadow },
            s(d),
            s(g),
            e.inputClass,
            s(T)
          ]],
          placeholder: s(l),
          type: s(n),
          disabled: s(a),
          readonly: s(o)
        }, y.$attrs), null, 16, fo), [
          [xt, s(f)]
        ]),
        R(y.$slots, "append.outer", {}, () => [
          m("div", {
            class: I(["absolute inset-y-0 right-0 h-full flex items-center px-3 text-gray-500", e.appendClass]),
            onClick: j[4] || (j[4] = (x) => t("clickAppend"))
          }, [
            R(y.$slots, "append", {}, () => [
              O(s(ze), {
                icon: e.appendIcon,
                class: I(["w-5 h-5", e.appendIconClass]),
                onClick: j[3] || (j[3] = (x) => t("clickAppendIcon"))
              }, null, 8, ["icon", "class"])
            ])
          ], 2)
        ])
      ])),
      s(p) ? (h(), w("div", po, C(s(p)), 1)) : B("", !0)
    ], 2));
  }
}), yo = { class: "relative" }, bo = { class: "block flex-grow w-full truncate mr-2" }, mo = { class: "w-auto" }, vo = {
  key: 0,
  class: "px-2 border-b py-2"
}, xo = {
  key: 1,
  class: "px-4 pb-2 pt-3"
}, wo = /* @__PURE__ */ q(" No results "), jo = /* @__PURE__ */ N({
  __name: "VSelect",
  props: {
    value: {
      type: [Object, String, Number, Boolean],
      default: ""
    },
    modelValue: {
      type: [Object, String, Number, Boolean],
      default: ""
    },
    items: {
      type: Array,
      default: () => []
    },
    color: {
      type: String,
      default: "primary"
    },
    placeholder: {
      type: String,
      default: "Select"
    },
    searchable: {
      type: Boolean,
      default: !1
    },
    hideCheckIcon: {
      type: Boolean,
      default: !1
    },
    btnClass: {
      type: String,
      default: ""
    },
    top: {
      type: Boolean,
      default: !1
    },
    itemText: {
      type: String,
      default: "text"
    },
    itemValue: {
      type: String,
      default: "value"
    },
    name: {
      type: String,
      default: ""
    },
    error: {
      type: Boolean,
      default: !1
    },
    errorMessages: {
      type: Array,
      default: () => []
    },
    returnObject: {
      type: Boolean,
      default: !1
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    clearText: {
      type: String,
      default: "Clear"
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    readonly: {
      type: Boolean,
      default: !1
    },
    label: {
      type: String,
      default: ""
    },
    labelClass: {
      type: String,
      default: "mb-1 block"
    },
    wrapperClass: {
      type: String,
      default: ""
    },
    errorClass: {
      type: String,
      default: "text-error-600 mt-1 text-sm"
    },
    rules: {
      type: String,
      default: ""
    },
    shadow: {
      type: Boolean,
      default: !1
    },
    shadowClass: {
      type: String,
      default: "shadow-sm"
    }
  },
  emits: ["update:modelValue", "update:value", "search"],
  setup(e, { emit: t }) {
    const r = e, {
      modelValue: n,
      color: o,
      items: a,
      placeholder: l,
      searchable: i,
      hideCheckIcon: c,
      btnClass: u,
      top: d,
      itemText: f,
      itemValue: p,
      value: g,
      returnObject: T,
      name: y,
      rules: j
    } = fe(r), x = Et(o.value), { value: v, errorMessage: U } = Be(y, j, {
      initialValue: n.value || g.value
    }), ee = S(() => U.value || r.errorMessages[0]);
    J(n, (b) => {
      v.value = b;
    }), J(g, (b) => {
      v.value = b;
    });
    const D = ie(""), k = S(() => {
      if (i.value) {
        const b = D.value.toLowerCase();
        return a.value.filter((E) => E[f.value]?.toLowerCase?.()?.includes(b) || E[p.value]?.toLowerCase?.()?.includes(b));
      } else
        return a.value;
    }), F = (b) => {
      if (T.value)
        v.value = b;
      else {
        const E = b || n.value || g.value;
        if (["string", "number", "boolean"].includes(typeof E)) {
          const V = k.value.find((Q) => Q[p.value] === E);
          V && (v.value = V);
        } else
          v.value = b;
      }
    };
    J(v, (b) => {
      const E = T.value ? b : b?.[p.value];
      t("update:modelValue", E), t("update:value", E);
    }), J(
      n,
      (b) => {
        F(b);
      },
      { immediate: !0 }
    ), J(
      g,
      (b) => {
        F(b);
      },
      { immediate: !0 }
    ), J(D, (b) => {
      t("search", b);
    });
    const H = S(() => v.value?.[f.value] || l.value), _ = () => v.value = "";
    return (b, E) => (h(), w("div", {
      class: I(e.wrapperClass)
    }, [
      O(s(St), {
        modelValue: s(v),
        "onUpdate:modelValue": E[1] || (E[1] = (V) => Qe(v) ? v.value = V : null)
      }, {
        default: A(() => [
          e.label ? (h(), P(s(Tt), {
            key: 0,
            class: I(e.labelClass)
          }, {
            default: A(() => [
              q(C(e.label), 1)
            ]),
            _: 1
          }, 8, ["class"])) : B("", !0),
          m("div", yo, [
            O(s(It), {
              class: I(["border relative w-full py-2 px-4 h-10 text-left bg-white rounded-md cursor-default focus:outline-none focus-visible:ring-2 focus-visible:ring-opacity-75 focus-visible:ring-white focus-visible:ring-offset-primary-300 focus-visible:ring-offset-2 focus-visible:border-primary-600 flex items-center gap-1", [
                s(u),
                e.error ? "border-error-500" : "",
                {
                  [e.shadowClass]: e.shadow
                }
              ]]),
              disabled: e.disabled
            }, {
              default: A(() => [
                m("div", bo, [
                  R(b.$slots, "selected", { item: s(v) }, () => [
                    q(C(s(H)), 1)
                  ])
                ]),
                s(v) && e.clearable ? (h(), P(s(_t), { key: 0 }, {
                  activator: A(({ on: V }) => [
                    m("span", $(wt(V), {
                      onClick: _,
                      class: "w-auto cursor-pointer"
                    }), [
                      O(s(Vt), {
                        class: "w-5 h-5 text-gray-400 hover:text-gray-500",
                        "aria-hidden": "true"
                      })
                    ], 16)
                  ]),
                  default: A(() => [
                    m("span", null, C(e.clearText), 1)
                  ]),
                  _: 1
                })) : B("", !0),
                m("span", mo, [
                  O(s(Ot), {
                    class: "w-5 h-5 text-gray-400",
                    "aria-hidden": "true"
                  })
                ])
              ]),
              _: 3
            }, 8, ["class", "disabled"]),
            O(kt, {
              "leave-active-class": "transition duration-100 ease-in",
              "leave-from-class": "opacity-100",
              "leave-to-class": "opacity-0"
            }, {
              default: A(() => [
                O(s(Bt), {
                  class: I(["absolute w-full py-1 mt-1 overflow-auto text-base bg-white rounded-md shadow-lg max-h-60 ring-1 ring-black ring-opacity-5 focus:outline-none sm:text-sm z-10", s(d) ? "bottom-10" : ""])
                }, {
                  default: A(() => [
                    s(i) ? (h(), w("div", vo, [
                      O(s(ho), {
                        modelValue: D.value,
                        "onUpdate:modelValue": E[0] || (E[0] = (V) => D.value = V),
                        size: "sm",
                        placeholder: "Search..."
                      }, null, 8, ["modelValue"])
                    ])) : B("", !0),
                    s(i) && !s(k).length ? (h(), w("div", xo, [
                      R(b.$slots, "empty", {}, () => [
                        wo
                      ])
                    ])) : B("", !0),
                    (h(!0), w(He, null, Ct(s(k), (V, Q) => (h(), P(s(jt), {
                      key: Q,
                      value: V,
                      as: "template"
                    }, {
                      default: A(({ active: te, selected: G }) => [
                        m("li", {
                          class: I(["group", [
                            te ? `text-white ${s(x)}` : "text-gray-900",
                            "cursor-default select-none relative py-2 pr-4",
                            s(c) ? "pl-4" : "pl-10"
                          ]])
                        }, [
                          m("span", {
                            class: I([
                              G ? "font-medium" : "font-normal",
                              "block truncate"
                            ])
                          }, [
                            R(b.$slots, "item", { item: V }, () => [
                              q(C(V?.[s(f)]), 1)
                            ])
                          ], 2),
                          G && !s(c) ? (h(), w("span", {
                            key: 0,
                            class: I(["absolute inset-y-0 left-0 flex items-center pl-3", [te ? "text-white" : "text-gray-900"]])
                          }, [
                            R(b.$slots, "icon", {}, () => [
                              O(s(At), {
                                class: "w-5 h-5",
                                "aria-hidden": "true"
                              })
                            ])
                          ], 2)) : B("", !0)
                        ], 2)
                      ]),
                      _: 2
                    }, 1032, ["value"]))), 128))
                  ]),
                  _: 3
                }, 8, ["class"])
              ]),
              _: 3
            })
          ])
        ]),
        _: 3
      }, 8, ["modelValue"]),
      s(ee) ? (h(), w("div", {
        key: 0,
        class: I(e.errorClass)
      }, C(s(ee)), 3)) : B("", !0)
    ], 2));
  }
});
export {
  jo as VSelect,
  jo as default
};
