// @gits-id/dropdown/dist/dropdown.mjs
import { defineComponent as b, computed as d, resolveComponent as B, openBlock as l, createBlock as s, unref as a, withCtx as n, createElementBlock as h, resolveDynamicComponent as p, mergeProps as f, renderSlot as r, normalizeClass as m, createTextVNode as g, toDisplayString as y, createElementVNode as k, createVNode as u, Transition as P, Fragment as x, renderList as D, normalizeProps as $, guardReactiveProps as I } from "vue";
import { MenuItem as T, Menu as M, MenuButton as C, MenuItems as V } from "@headlessui/vue";
import { Icon as z } from "@gits-id/icon";
const N = {
  key: 0,
  class: "border-b my-1 -mx-1"
}, E = /* @__PURE__ */ b({
  __name: "DropdownItem",
  props: {
    text: null,
    to: null,
    href: null,
    icon: null,
    iconClass: { default: "w-5 h-5" },
    newTab: { type: Boolean },
    divider: { type: Boolean }
  },
  setup(e) {
    const t = e, i = d(() => t.href ? "a" : t.to ? B("RouterLink") : "button"), o = d(() => t.to ? { to: t.to } : {}), w = d(() => t.href ? { href: t.href } : {});
    return (c, L) => (l(), s(a(T), null, {
      default: n(({ active: v }) => [
        e.divider ? (l(), h("div", N)) : (l(), s(p(a(i)), f({
          key: 1,
          target: e.href && e.newTab ? "_blank" : void 0,
          rel: e.href && e.newTab ? "noopener" : null,
          class: [
            v ? "bg-gray-200" : "text-gray-900",
            "group flex gap-2 rounded-md items-center w-full px-2 py-2"
          ]
        }, { ...a(w), ...a(o), ...c.$attrs }), {
          default: n(() => [
            r(c.$slots, "icon", {}, () => [
              (l(), s(p(e.icon), {
                class: m(e.iconClass)
              }, null, 8, ["class"]))
            ]),
            r(c.$slots, "default", {}, () => [
              g(y(e.text), 1)
            ])
          ]),
          _: 2
        }, 1040, ["target", "rel", "class"]))
      ]),
      _: 3
    }));
  }
}), F = /* @__PURE__ */ b({
  __name: "Dropdown",
  props: {
    modelValue: { type: Boolean, default: !1 },
    btnProps: { default: () => ({
      variant: "secondary"
    }) },
    label: { default: "Options" },
    right: { type: Boolean, default: !1 },
    items: { default: () => [] },
    top: { type: Boolean, default: !1 },
    topClass: { default: "bottom-10" },
    bottomClass: { default: "top-6" },
    panelClass: { default: "" },
    buttonWrapperClass: { default: "" },
    rightClass: { default: "right-0" },
    leftClass: { default: "left-0" }
  },
  setup(e) {
    return (t, i) => (l(), s(a(M), {
      as: "div",
      class: "relative inline-block text-left"
    }, {
      default: n(() => [
        k("div", {
          class: m(e.buttonWrapperClass)
        }, [
          r(t.$slots, "activator", {
            btnProps: e.btnProps,
            label: e.label
          }, () => [
            u(a(C), f({ as: "button" }, e.btnProps), {
              default: n(() => [
                g(y(e.label) + " ", 1),
                u(a(z), {
                  name: "ri:arrow-down-s-line",
                  class: "w-5 h-5 inline",
                  "aria-hidden": "true"
                })
              ]),
              _: 1
            }, 16)
          ])
        ], 2),
        u(P, {
          "enter-active-class": "transition duration-100 ease-out",
          "enter-from-class": "transform scale-95 opacity-0",
          "enter-to-class": "transform scale-100 opacity-100",
          "leave-active-class": "transition duration-75 ease-in",
          "leave-from-class": "transform scale-100 opacity-100",
          "leave-to-class": "transform scale-95 opacity-0"
        }, {
          default: n(() => [
            u(a(V), {
              class: m(["absolute z-10 p-1 w-56 mt-2 bg-white rounded-md shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none", [
                e.right ? e.rightClass : e.leftClass,
                e.top ? e.topClass : e.bottomClass,
                e.panelClass
              ]])
            }, {
              default: n(() => [
                r(t.$slots, "default", {}, () => [
                  (l(!0), h(x, null, D(e.items, (o) => (l(), s(E, f({
                    key: o.text
                  }, o), null, 16))), 128))
                ])
              ]),
              _: 3
            }, 8, ["class"])
          ]),
          _: 3
        })
      ]),
      _: 3
    }));
  }
}), O = /* @__PURE__ */ b({
  __name: "DropdownButton",
  setup(e) {
    return (t, i) => (l(), s(a(C), null, {
      default: n((o) => [
        r(t.$slots, "default", $(I(o)))
      ]),
      _: 3
    }));
  }
});
export {
  F as Dropdown,
  O as DropdownButton,
  E as DropdownItem,
  F as default
};
