// @gits-id/menus/dist/menus.mjs
import { defineComponent as g, computed as v, openBlock as c, createBlock as m, unref as t, withCtx as n, createElementVNode as r, createVNode as s, normalizeClass as u, renderSlot as a, Transition as h, createElementBlock as _, Fragment as k, renderList as w, toDisplayString as C, createCommentVNode as b } from "vue";
import { ChevronDownIcon as x, ChevronRightIcon as B } from "@heroicons/vue/solid";
import { Menu as I, MenuButton as M, MenuItems as V, MenuItem as $ } from "@headlessui/vue";
import N from "@gits-id/button";
const S = (e, i) => {
  const l = e.__vccOpts || e;
  for (const [o, d] of i)
    l[o] = d;
  return l;
}, j = { class: "px-1 py-1" }, z = { class: "flex-none" }, D = /* @__PURE__ */ g({
  __name: "VMenus",
  props: {
    items: {
      type: Array,
      default: () => []
    },
    right: {
      type: Boolean,
      default: !1
    },
    small: {
      type: Boolean,
      default: !1
    },
    rightIcon: {
      type: Boolean,
      default: !1
    },
    btnClass: {
      type: String,
      default: ""
    }
  },
  setup(e) {
    const i = e, l = v(() => i.right ? "right-0" : "left-0");
    return (o, d) => (c(), m(t(I), {
      as: "div",
      class: "relative inline-block mx-4 text-left"
    }, {
      default: n(() => [
        r("div", null, [
          s(t(M), {
            class: u(["inline-flex font-medium text-gray-700 justify-center items-center w-full py-1 focus:outline-none", [e.btnClass, e.small ? "text-sm" : ""]])
          }, {
            default: n(() => [
              a(o.$slots, "default", {}, void 0, !0),
              a(o.$slots, "icon", {}, () => [
                s(t(x), {
                  class: "w-5 h-5 ml-2 -mr-1",
                  "aria-hidden": "true"
                })
              ], !0)
            ]),
            _: 3
          }, 8, ["class"])
        ]),
        s(h, {
          "enter-active-class": "transition duration-100 ease-out",
          "enter-from-class": "transform scale-95 opacity-0",
          "enter-to-class": "transform scale-100 opacity-100",
          "leave-active-class": "transition duration-75 ease-in",
          "leave-from-class": "transform scale-100 opacity-100",
          "leave-to-class": "transform scale-95 opacity-0"
        }, {
          default: n(() => [
            s(t(V), {
              class: u(["absolute z-10 w-56 mt-2 bg-white divide-y divide-gray-100 rounded-md shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none", t(l)])
            }, {
              default: n(() => [
                r("div", {
                  class: u(["arrow", { right: e.right }])
                }, null, 2),
                r("div", j, [
                  a(o.$slots, "prepend", {}, void 0, !0),
                  (c(!0), _(k, null, w(e.items, (f, p) => (c(), m(t($), { key: p }, {
                    default: n(({ active: y }) => [
                      s(t(N), {
                        to: f.to,
                        "no-ring": "",
                        block: "",
                        color: y ? "primary" : "",
                        class: "rounded border-none !justify-start font-normal"
                      }, {
                        default: n(() => [
                          r("span", z, C(f.text), 1),
                          e.rightIcon ? a(o.$slots, "right-icon", { key: 0 }, () => [
                            s(t(B), { class: "w-6 h-6" })
                          ], !0) : b("", !0)
                        ]),
                        _: 2
                      }, 1032, ["to", "color"])
                    ]),
                    _: 2
                  }, 1024))), 128)),
                  a(o.$slots, "append", {}, void 0, !0)
                ])
              ]),
              _: 3
            }, 8, ["class"])
          ]),
          _: 3
        })
      ]),
      _: 3
    }));
  }
}), O = /* @__PURE__ */ S(D, [["__scopeId", "data-v-1ce53d60"]]);
export {
  O as VMenus,
  O as default
};
