// @gits-id/card/dist/card.mjs
import { defineComponent as y, toRefs as C, computed as c, openBlock as d, createBlock as m, resolveDynamicComponent as b, unref as r, normalizeClass as a, withCtx as g, renderSlot as l, createElementBlock as s, createElementVNode as f, toDisplayString as S, createCommentVNode as n } from "vue";
const B = /* @__PURE__ */ y({
  __name: "VCard",
  props: {
    title: {
      type: String,
      default: ""
    },
    defaultWrapperClass: {
      type: String,
      default: "rounded-md bg-white flex flex-col"
    },
    defaultHeaderClass: {
      type: String,
      default: "font-semibold flex px-4"
    },
    defaultFooterClass: {
      type: String,
      default: "px-4 flex"
    },
    defaultBodyClass: {
      type: String,
      default: "px-4 flex flex-col"
    },
    wrapperClass: {
      type: String,
      default: ""
    },
    headerClass: {
      type: String,
      default: ""
    },
    footerClass: {
      type: String,
      default: ""
    },
    bodyClass: {
      type: String,
      default: ""
    },
    hideHeader: {
      type: Boolean,
      default: !1
    },
    hideFooter: {
      type: Boolean,
      default: !1
    },
    bordered: {
      type: Boolean,
      default: !1
    },
    flat: {
      type: Boolean,
      default: !1
    },
    to: {
      type: String,
      default: ""
    }
  },
  setup(e) {
    const u = e, { to: o } = C(u), i = c(() => o.value ? "router-link" : "div");
    return (t, h) => (d(), m(b(r(i)), {
      to: r(o),
      class: a([
        e.defaultWrapperClass,
        e.wrapperClass,
        {
          border: e.bordered,
          shadow: !e.flat
        }
      ])
    }, {
      default: g(() => [
        l(t.$slots, "image"),
        e.hideHeader ? n("", !0) : (d(), s("div", {
          key: 0,
          class: a([
            e.defaultHeaderClass,
            e.bordered ? "border-b py-3" : "pt-3",
            e.headerClass
          ])
        }, [
          l(t.$slots, "header", {}, () => [
            f("div", null, S(e.title), 1)
          ])
        ], 2)),
        f("div", {
          class: a([
            e.defaultBodyClass,
            e.bordered ? "border-t py-3" : "py-3",
            e.bodyClass
          ])
        }, [
          l(t.$slots, "default")
        ], 2),
        e.hideFooter ? n("", !0) : (d(), s("div", {
          key: 1,
          class: a([
            e.defaultFooterClass,
            e.bordered ? "border-t py-3" : "pb-3",
            e.footerClass
          ])
        }, [
          l(t.$slots, "footer")
        ], 2))
      ]),
      _: 3
    }, 8, ["to", "class"]));
  }
});
export {
  B as VCard,
  B as default
};
