// @gits-id/avatar/dist/avatar.mjs
import { defineComponent as d, computed as a, openBlock as l, createElementBlock as n, normalizeClass as p, unref as s, normalizeStyle as z, mergeProps as f, toDisplayString as h } from "vue";
const r = ["xxs", "xs", "sm", "md", "lg", "xl"];
const v = ["src", "alt"], x = {
  key: 1,
  class: "avatar-initial"
}, y = d({
  __name: "Avatar",
  props: {
    color: null,
    size: { default: "md" },
    shape: { default: "circle" },
    alt: null,
    src: null,
    name: null,
    maxInitial: { default: 2 }
  },
  setup(t) {
    const e = t, c = a(() => [
      `avatar avatar-${e.color} avatar-${e.size} avatar-${e.shape}`,
      {}
    ]), o = a(() => e.name?.split(" ").map((i) => i[0]).slice(0, e.maxInitial).join("")), u = a(() => r.includes(String(e.size)) ? {} : {
      width: e.size,
      height: e.size
    }), m = a(() => r.includes(String(e.size)) ? {} : {
      width: `${e.size}px`,
      height: `${e.size}px`
    });
    return (i, g) => (l(), n("div", {
      class: p(s(c)),
      style: z(s(m))
    }, [
      t.src ? (l(), n("img", f({
        key: 0,
        src: t.src,
        alt: t.name || t.alt,
        class: "avatar-image"
      }, s(u)), null, 16, v)) : (l(), n("span", x, h(s(o)), 1))
    ], 6));
  }
});
export {
  y as default
};
