// @gits-id/ui/dist/ui.mjs
import u from "@gits-id/button";
import { default as Xo } from "@gits-id/button";
import d from "@gits-id/alert";
import { default as ot } from "@gits-id/alert";
import x from "@gits-id/spinner";
import { default as et } from "@gits-id/spinner";
import g from "@gits-id/app-bar";
import { default as nt } from "@gits-id/app-bar";
import h from "@gits-id/badge";
import { default as at } from "@gits-id/badge";
import I from "@gits-id/breadcrumbs";
import { default as st } from "@gits-id/breadcrumbs";
import D from "@gits-id/card";
import { default as Vt } from "@gits-id/card";
import { VCheckbox as L, VFormSelect as S, VInput as b, VInputGroup as v, VInputRange as w, VRadio as B, VRadioGroup as C, VTextarea as T, VFormGroup as y, VFileUpload as A } from "@gits-id/forms";
import { VCheckbox as pt, VFileUpload as ct, VFormGroup as ut, VFormSelect as dt, VInput as xt, VInputGroup as gt, VInputRange as ht, VRadio as It, VRadioGroup as Dt, VTextarea as Lt } from "@gits-id/forms";
import { VDataTable as G, VDataTablePagination as F } from "@gits-id/table";
import { VDataTable as bt, VDataTablePagination as vt } from "@gits-id/table";
import { Dropdown as M, DropdownItem as P, DropdownButton as R } from "@gits-id/dropdown";
import { Dropdown as Bt, DropdownButton as Ct, DropdownItem as Tt } from "@gits-id/dropdown";
import z from "@gits-id/modal";
import { default as At } from "@gits-id/modal";
import k from "@gits-id/pagination";
import { default as Ft } from "@gits-id/pagination";
import N from "@gits-id/progress-bar";
import { default as Pt } from "@gits-id/progress-bar";
import $ from "@gits-id/select";
import { default as zt } from "@gits-id/select";
import _ from "@gits-id/switch";
import { default as Nt } from "@gits-id/switch";
import H from "@gits-id/tabs";
import { default as _t } from "@gits-id/tabs";
import U from "@gits-id/tooltip";
import { default as Ut } from "@gits-id/tooltip";
import j from "@gits-id/navbar";
import { default as Et } from "@gits-id/navbar";
import E from "@gits-id/nav-drawer";
import { default as qt } from "@gits-id/nav-drawer";
import Z from "@gits-id/menu";
import { default as Kt } from "@gits-id/menu";
import q from "@gits-id/menus";
import { default as Qt } from "@gits-id/menus";
import J from "@gits-id/container";
import { default as Xt } from "@gits-id/container";
import K from "@gits-id/logo";
import { default as oe } from "@gits-id/logo";
import O from "@gits-id/toast";
export * from "@gits-id/toast";
import { default as re } from "@gits-id/toast";
import Q from "@gits-id/stats";
import { default as me } from "@gits-id/stats";
import { VCollapse as W, VCollapsible as X, VCollapsibleGroup as Y } from "@gits-id/collapsible";
import { VCollapse as ie, VCollapsible as se, VCollapsibleGroup as le } from "@gits-id/collapsible";
import oo from "@gits-id/multi-select";
import { default as fe } from "@gits-id/multi-select";
import { Auth as ce, Base as ue, Default as de } from "@gits-id/layouts";
export * from "@gits-id/pages";
import { Activation as he, Auth as Ie, ForgotPassword as De, Login as Le, NotFound as Se, ResetPassword as be } from "@gits-id/pages";
import { List as to, ListItem as eo, ListItemDivider as ro, ListItemHeader as no, ListCollapse as mo } from "@gits-id/list";
import { List as we, ListCollapse as Be, ListItem as Ce, ListItemDivider as Te, ListItemHeader as ye } from "@gits-id/list";
import { Icon as ao } from "@gits-id/icon";
import { Icon as Ge } from "@gits-id/icon";
import { defineComponent as f, computed as m, openBlock as r, createElementBlock as n, normalizeClass as io, unref as a, normalizeStyle as so, mergeProps as p, toDisplayString as lo, Fragment as Vo, renderList as fo } from "vue";
export * from "@gits-id/utils";
export * from "@gits-id/services";
const V = ["xxs", "xs", "sm", "md", "lg", "xl"], po = ["src", "alt"], co = {
  key: 1,
  class: "avatar-initial"
}, uo = f({
  __name: "Avatar",
  props: {
    color: null,
    size: { default: "md" },
    shape: { default: "circle" },
    alt: null,
    src: null,
    name: null,
    maxInitial: { default: 2 }
  },
  setup(o) {
    const t = o, e = m(() => [
      `avatar avatar-${t.color} avatar-${t.size} avatar-${t.shape}`,
      {}
    ]), s = m(() => t.name?.split(" ").map((l) => l[0]).slice(0, t.maxInitial).join("")), i = m(() => V.includes(String(t.size)) ? {} : {
      width: t.size,
      height: t.size
    }), c = m(() => V.includes(String(t.size)) ? {} : {
      width: `${t.size}px`,
      height: `${t.size}px`
    });
    return (l, go) => (r(), n("div", {
      class: io(a(e)),
      style: so(a(c))
    }, [
      o.src ? (r(), n("img", p({
        key: 0,
        src: o.src,
        alt: o.name || o.alt,
        class: "avatar-image"
      }, a(i)), null, 16, po)) : (r(), n("span", co, lo(a(s)), 1))
    ], 6));
  }
}), xo = f({
  __name: "Shimmer",
  props: {
    width: { default: 100 },
    height: { default: 16 },
    rounded: { type: Boolean },
    lines: { default: 1 }
  },
  setup(o) {
    const t = (e) => typeof e == "number" ? `${e}px` : e;
    return (e, s) => (r(!0), n(Vo, null, fo(o.lines, (i) => (r(), n("div", p({
      key: i,
      class: ["shimmer", { "shimmer--rounded": o.rounded }],
      style: { width: t(o.width), height: t(o.height) }
    }, e.$attrs), null, 16))), 128));
  }
});
function Ko() {
  return {
    type: "component",
    resolve: (o) => {
      if (o.match(/^V[A-Z]/))
        return { name: o, from: "@gits-id/ui" };
    }
  };
}
const Oo = {
  install: (o) => {
    o.component("VAlert", d), o.component("VBtn", u), o.component("VAppBar", g), o.component("VBadge", h), o.component("VBreadcrumbs", I), o.component("VCard", D), o.component("VCheckbox", L), o.component("VDataTable", G), o.component("VDataTablePagination", F), o.component("VDropdown", M), o.component("VDropdownItem", P), o.component("VDropdownButton", R), o.component("VFormSelect", S), o.component("VInput", b), o.component("VInputGroup", v), o.component("VInputRange", w), o.component("VModal", z), o.component("VPagination", k), o.component("VProgressBar", N), o.component("VRadio", B), o.component("VRadioGroup", C), o.component("VSelect", $), o.component("VSwitch", _), o.component("VTabs", H), o.component("VTextarea", T), o.component("VTooltip", U), o.component("VFormGroup", y), o.component("VNavbar", j), o.component("VNavDrawer", E), o.component("VMenu", Z), o.component("VMenus", q), o.component("VContainer", J), o.component("VToast", O), o.component("VSpinner", x), o.component("VStats", Q), o.component("VFileUpload", A), o.component("VCollapse", W), o.component("VCollapsible", X), o.component("VCollapsibleGroup", Y), o.component("VMultiSelect", oo), o.component("VLogo", K), o.component("VList", to), o.component("VListItem", eo), o.component("VListItemDivider", ro), o.component("VListItemHeader", no), o.component("VListCollapse", mo), o.component("VIcon", ao), o.component("VAvatar", uo), o.component("VShimmer", xo);
  }
};
export {
  he as Activation,
  Ie as Auth,
  ce as AuthLayout,
  ue as BaseLayout,
  de as DefaultLayout,
  De as ForgotPassword,
  Ko as GITSUIResolver,
  Le as Login,
  Se as NotFound,
  be as ResetPassword,
  ot as VAlert,
  nt as VAppBar,
  uo as VAvatar,
  at as VBadge,
  st as VBreadcrumbs,
  Xo as VBtn,
  Vt as VCard,
  pt as VCheckbox,
  ie as VCollapse,
  se as VCollapsible,
  le as VCollapsibleGroup,
  Xt as VContainer,
  bt as VDataTable,
  vt as VDataTablePagination,
  Bt as VDropdown,
  Ct as VDropdownButton,
  Tt as VDropdownItem,
  ct as VFileUpload,
  ut as VFormGroup,
  dt as VFormSelect,
  Ge as VIcon,
  xt as VInput,
  gt as VInputGroup,
  ht as VInputRange,
  we as VList,
  Be as VListCollapse,
  Ce as VListItem,
  Te as VListItemDivider,
  ye as VListItemHeader,
  oe as VLogo,
  Kt as VMenu,
  Qt as VMenus,
  At as VModal,
  fe as VMultiSelect,
  qt as VNavDrawer,
  Et as VNavbar,
  Ft as VPagination,
  Pt as VProgressBar,
  It as VRadio,
  Dt as VRadioGroup,
  zt as VSelect,
  xo as VShimmer,
  et as VSpinner,
  me as VStats,
  Nt as VSwitch,
  _t as VTabs,
  Lt as VTextarea,
  re as VToast,
  Ut as VTooltip,
  Oo as default
};
