// @gits-id/modal/dist/modal.mjs
import { defineComponent as X, toRefs as E, ref as w, watch as y, openBlock as i, createElementBlock as p, Fragment as H, renderSlot as m, createVNode as n, unref as e, withCtx as a, createElementVNode as u, normalizeClass as r, createCommentVNode as c, createBlock as h, toDisplayString as v, mergeProps as x, createTextVNode as P } from "vue";
import { TransitionRoot as R, Dialog as S, TransitionChild as T, DialogPanel as A, DialogTitle as L } from "@headlessui/vue";
import C from "@gits-id/button";
import { XIcon as O } from "@heroicons/vue/outline";
const q = { class: "fixed inset-0 z-30 overflow-y-auto" }, G = /* @__PURE__ */ u("div", { class: "fixed bg-black bg-opacity-50 inset-0" }, null, -1), J = {
  key: 0,
  class: "inline-block h-screen align-middle",
  "aria-hidden": "true"
}, K = { class: "text-lg font-medium leading-6 text-gray-900" }, Q = {
  inheritAttrs: !1
}, ee = /* @__PURE__ */ X({
  ...Q,
  __name: "VModal",
  props: {
    modelValue: { type: Boolean, default: !1 },
    title: { default: "" },
    confirm: { type: Boolean, default: !1 },
    confirmColor: { default: "primary" },
    confirmProps: { default: () => ({}) },
    confirmText: { default: "Confirm" },
    closeText: { default: "Close" },
    closeProps: { default: () => ({}) },
    headerClass: { default: "" },
    bodyClass: { default: "" },
    footerClass: { default: "" },
    modalClass: { default: "" },
    loading: { type: Boolean, default: !1 },
    hideHeader: { type: Boolean, default: !1 },
    hideFooter: { type: Boolean, default: !1 },
    centered: { type: Boolean, default: !1 },
    fullscreen: { type: Boolean, default: !1 },
    persistent: { type: Boolean, default: !1 },
    hideXButton: { type: Boolean, default: !1 },
    xButtonProps: { default: () => ({}) },
    xIconClass: { default: "" }
  },
  emits: ["update:modelValue", "confirm", "close", "open"],
  setup(l, { emit: s }) {
    const g = l, {
      modelValue: B,
      title: M,
      confirm: D,
      confirmColor: N,
      confirmProps: k,
      confirmText: $,
      closeText: j,
      closeProps: z,
      loading: F
    } = E(g), d = w(B.value), o = w(F.value);
    y(B, (t) => d.value = t), y(d, (t) => s("update:modelValue", t)), y(o, (t) => o.value = t);
    function I() {
      if (g.persistent || o.value)
        return null;
      f();
    }
    function f() {
      d.value = !1, s("update:modelValue", !1), s("close");
    }
    function b() {
      d.value = !0, s("update:modelValue", !0), s("open");
    }
    const V = () => {
      s("confirm", {
        open: b,
        close: f
      });
    };
    return (t, U) => (i(), p(H, null, [
      m(t.$slots, "activator", { open: b }),
      n(e(R), {
        appear: "",
        show: d.value,
        as: "template"
      }, {
        default: a(() => [
          n(e(S), {
            static: "",
            open: d.value,
            as: "div",
            onClose: I
          }, {
            default: a(() => [
              u("div", q, [
                u("div", {
                  class: r(["min-h-screen text-center", { "px-4": !l.fullscreen }])
                }, [
                  n(e(T), {
                    as: "template",
                    enter: "duration-300 ease-out",
                    "enter-from": "opacity-0",
                    "enter-to": "opacity-100",
                    leave: "duration-200 ease-in",
                    "leave-from": "opacity-100",
                    "leave-to": "opacity-0"
                  }, {
                    default: a(() => [
                      G
                    ]),
                    _: 1
                  }),
                  l.fullscreen ? c("", !0) : (i(), p("span", J, " \u200B ")),
                  n(e(T), {
                    as: "template",
                    enter: "duration-300 ease-out",
                    "enter-from": "opacity-0 scale-95",
                    "enter-to": "opacity-100 scale-100",
                    leave: "duration-200 ease-in",
                    "leave-from": "opacity-100 scale-100",
                    "leave-to": "opacity-0 scale-95"
                  }, {
                    default: a(() => [
                      n(e(A), {
                        class: r(["w-full overflow-hidden align-middle transition-all transform bg-white shadow-xl p-6", [
                          {
                            "max-w-md inline-block rounded-lg my-8": !l.fullscreen,
                            "h-screen flex flex-col": l.fullscreen
                          },
                          l.centered ? "text-center" : "text-left",
                          l.modalClass
                        ]])
                      }, {
                        default: a(() => [
                          l.hideHeader ? c("", !0) : (i(), h(e(L), {
                            key: 0,
                            as: "div",
                            class: r(["flex justify-between gap-4 items-center", l.headerClass])
                          }, {
                            default: a(() => [
                              m(t.$slots, "header", {}, () => [
                                u("h3", K, v(e(M)), 1)
                              ]),
                              l.hideXButton ? c("", !0) : (i(), h(e(C), x({
                                key: 0,
                                icon: "",
                                text: "",
                                rounded: "",
                                size: "sm",
                                class: "p-0",
                                disabled: o.value
                              }, l.xButtonProps, { onClick: f }), {
                                default: a(() => [
                                  n(e(O), {
                                    class: r(["w-5 h-5", l.xIconClass])
                                  }, null, 8, ["class"])
                                ]),
                                _: 1
                              }, 16, ["disabled"]))
                            ]),
                            _: 3
                          }, 8, ["class"])),
                          u("div", {
                            class: r(["mt-4 text-gray-600 flex-1", [l.centered ? "text-center" : "", l.bodyClass]])
                          }, [
                            m(t.$slots, "default")
                          ], 2),
                          l.hideFooter ? c("", !0) : (i(), p("div", {
                            key: 1,
                            class: r(["mt-6 flex gap-2", [
                              l.centered ? "justify-center" : "justify-end",
                              l.footerClass
                            ]])
                          }, [
                            m(t.$slots, "footer", {
                              loading: o.value,
                              confirmProps: e(k),
                              onConfirm: V
                            }, () => [
                              e(D) ? (i(), h(e(C), x({
                                key: 0,
                                color: e(N),
                                loading: o.value
                              }, e(k), { onClick: V }), {
                                default: a(() => [
                                  P(v(e($)), 1)
                                ]),
                                _: 1
                              }, 16, ["color", "loading"])) : c("", !0),
                              n(e(C), x(e(z), {
                                disabled: o.value,
                                onClick: f
                              }), {
                                default: a(() => [
                                  P(v(e(j)), 1)
                                ]),
                                _: 1
                              }, 16, ["disabled"])
                            ])
                          ], 2))
                        ]),
                        _: 3
                      }, 8, ["class"])
                    ]),
                    _: 3
                  })
                ], 2)
              ])
            ]),
            _: 3
          }, 8, ["open"])
        ]),
        _: 3
      }, 8, ["show"])
    ], 64));
  }
});
export {
  ee as VModal,
  ee as default
};
