// @gits-id/navbar/dist/navbar.mjs
import { defineComponent as F, toRefs as O, ref as T, watch as $, computed as A, resolveComponent as R, openBlock as o, createElementBlock as i, normalizeClass as g, createElementVNode as a, createVNode as s, unref as e, withCtx as t, renderSlot as N, Transition as U, Fragment as m, renderList as _, createBlock as c, toDisplayString as d, createTextVNode as r, withModifiers as B, createCommentVNode as q, mergeProps as G } from "vue";
import { ChevronDownIcon as h, UserCircleIcon as D } from "@heroicons/vue/solid";
import { MenuIcon as H } from "@heroicons/vue/outline";
import f from "@gits-id/button";
import { Disclosure as L, DisclosureButton as M, DisclosurePanel as j, Menu as J, MenuItem as K } from "@headlessui/vue";
import z from "@gits-id/menus";
import I from "@gits-id/logo";
const Q = { class: "container mx-auto sm:hidden" }, W = { class: "flex w-full items-center px-4 justify-between" }, X = /* @__PURE__ */ a("div", null, null, -1), Y = {
  key: 0,
  class: "shadow-lg mt-1 absolute bg-white w-full z-10"
}, Z = {
  class: "my-3 block w-full",
  "aria-label": "Navigation"
}, ee = { class: "mt-2" }, te = { class: "inline-flex items-center" }, le = /* @__PURE__ */ r(" Logout "), se = {
  key: 1,
  class: "px-4"
}, oe = /* @__PURE__ */ r("Login"), ae = { class: "container mx-auto hidden sm:flex gap-x-4 items-center px-4 2xl:px-0" }, ne = {
  class: "ml-4 flex items-center flex-grow",
  "aria-label": "Desktop Navigation"
}, ie = { class: "p-0.5" }, ce = /* @__PURE__ */ r(" Login "), ge = /* @__PURE__ */ F({
  __name: "VNavbar",
  props: {
    modelValue: {
      type: Boolean,
      default: !1
    },
    small: {
      type: Boolean,
      default: !1
    },
    user: {
      type: Object,
      default: null
    },
    menus: {
      type: Array,
      default: () => []
    },
    loginPath: {
      type: String,
      default: ""
    },
    dense: {
      type: Boolean,
      default: !1
    },
    imgClass: {
      type: String,
      default: "h-7"
    }
  },
  emits: ["update:modelValue", "logout"],
  setup(n, { emit: v }) {
    const k = n, { menus: b, modelValue: P, user: p, loginPath: w } = O(k), u = T(k.modelValue);
    $(
      P,
      (x) => {
        u.value = x;
      },
      { immediate: !0 }
    );
    const S = () => {
      u.value = !u.value, v("update:modelValue", u.value);
    }, C = () => {
      v("logout");
    }, E = A(() => p.value?.name?.split(" ")?.[0]);
    return (x, re) => {
      const V = R("router-link");
      return o(), i("div", {
        class: g(["shadow-md", [n.dense ? "py-1" : "py-2"]])
      }, [
        a("div", Q, [
          a("div", W, [
            s(e(f), {
              rounded: "",
              icon: "",
              text: "",
              onClick: S
            }, {
              default: t(() => [
                s(e(H), { class: "w-6 h-6" })
              ]),
              _: 1
            }),
            N(x.$slots, "logo", {}, () => [
              s(e(I), { "img-class": n.imgClass }, null, 8, ["img-class"])
            ]),
            X
          ]),
          s(U, {
            "enter-active-class": "duration-150 ease-out",
            "enter-from-class": "opacity-0 scale-95",
            "enter-to-class": "opacity-100 scale-100",
            "leave-active-class": "duration-100 ease-in",
            "leave-from-class": "opacity-100 scale-100",
            "leave-to-class": "opacity-0 scale-95"
          }, {
            default: t(() => [
              u.value ? (o(), i("div", Y, [
                a("nav", Z, [
                  (o(!0), i(m, null, _(e(b), (l) => (o(), i(m, {
                    key: l.text
                  }, [
                    l.children ? (o(), c(e(L), { key: 0 }, {
                      default: t(() => [
                        s(e(M), { class: "flex items-center justify-between w-full text-left font-semibold py-2 px-4 text-gray-700 focus:outline-none" }, {
                          default: t(() => [
                            a("span", null, d(l.text), 1),
                            s(e(h), { class: "w-5 h-5" })
                          ]),
                          _: 2
                        }, 1024),
                        s(e(j), { class: "px-4 pb-2 text-gray-600" }, {
                          default: t(() => [
                            (o(!0), i(m, null, _(l.children, (y) => (o(), c(V, {
                              key: y.text,
                              to: y.to,
                              class: "font-semibold block py-2 px-4 text-gray-700"
                            }, {
                              default: t(() => [
                                r(d(y.text), 1)
                              ]),
                              _: 2
                            }, 1032, ["to"]))), 128))
                          ]),
                          _: 2
                        }, 1024)
                      ]),
                      _: 2
                    }, 1024)) : (o(), c(V, {
                      key: 1,
                      to: l.to,
                      class: "font-semibold block py-2 px-4"
                    }, {
                      default: t(() => [
                        r(d(l.text), 1)
                      ]),
                      _: 2
                    }, 1032, ["to"]))
                  ], 64))), 128)),
                  a("div", ee, [
                    e(p) ? (o(), c(e(L), { key: 0 }, {
                      default: t(() => [
                        s(e(M), { class: "flex items-center justify-between w-full px-4 py-2 text-left font-semibold block py-2 px-4 text-gray-700 focus:outline-none" }, {
                          default: t(() => [
                            a("span", te, [
                              s(e(D), { class: "w-[30px] inline h-[30px] mr-2" }),
                              a("span", null, d(e(p).name), 1)
                            ]),
                            s(e(h), { class: "w-5 h-5" })
                          ]),
                          _: 1
                        }),
                        s(e(j), { class: "px-4 pt-2" }, {
                          default: t(() => [
                            s(e(f), {
                              color: "primary",
                              outlined: "",
                              dense: "",
                              block: "",
                              onClick: B(C, ["prevent"])
                            }, {
                              default: t(() => [
                                le
                              ]),
                              _: 1
                            }, 8, ["onClick"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    })) : (o(), i("div", se, [
                      s(e(f), {
                        to: e(w),
                        color: "primary",
                        rounded: "",
                        block: "",
                        outlined: "",
                        dense: "",
                        solid: ""
                      }, {
                        default: t(() => [
                          oe
                        ]),
                        _: 1
                      }, 8, ["to"])
                    ]))
                  ])
                ])
              ])) : q("", !0)
            ]),
            _: 1
          })
        ]),
        a("div", ae, [
          N(x.$slots, "logo", {}, () => [
            s(e(I), { "img-class": n.imgClass }, null, 8, ["img-class"])
          ]),
          a("nav", ne, [
            (o(!0), i(m, null, _(e(b), (l) => (o(), i(m, {
              key: l.text
            }, [
              l.children ? (o(), c(e(z), {
                key: 0,
                items: l.children,
                small: n.small
              }, {
                default: t(() => [
                  r(d(l.text), 1)
                ]),
                _: 2
              }, 1032, ["items", "small"])) : (o(), c(e(f), G({ key: 1 }, l, {
                to: l.to,
                size: l.small || n.small ? "sm" : "",
                class: l.small || n.small ? "text-sm" : ""
              }), {
                default: t(() => [
                  r(d(l.text), 1)
                ]),
                _: 2
              }, 1040, ["to", "size", "class"]))
            ], 64))), 128))
          ]),
          e(p) ? (o(), c(e(J), {
            key: 0,
            as: "div",
            class: "relative inline-block text-left mt-1"
          }, {
            default: t(() => [
              s(e(z), { right: "" }, {
                append: t(() => [
                  a("div", ie, [
                    s(e(K), {
                      onClick: B(C, ["prevent"])
                    }, {
                      default: t(({ active: l }) => [
                        a("button", {
                          class: g([
                            l ? "bg-primary-50 text-primary-500" : "text-gray-700",
                            "group flex rounded-md items-center w-full px-3 py-2 font-semibold",
                            n.small ? "text-sm" : ""
                          ])
                        }, " Logout ", 2)
                      ]),
                      _: 1
                    }, 8, ["onClick"])
                  ])
                ]),
                icon: t(() => [
                  s(e(h), { class: "ml-2 w-5 h-5" })
                ]),
                default: t(() => [
                  s(e(D), {
                    class: g(["w-[30px] h-[30px] mr-2", [n.small ? "text-sm" : ""]])
                  }, null, 8, ["class"]),
                  r(" " + d(e(E)), 1)
                ]),
                _: 1
              })
            ]),
            _: 1
          })) : (o(), c(e(f), {
            key: 1,
            to: e(w),
            color: "primary",
            outlined: "",
            solid: "",
            wide: "",
            rounded: ""
          }, {
            default: t(() => [
              ce
            ]),
            _: 1
          }, 8, ["to"]))
        ])
      ], 2);
    };
  }
});
export {
  ge as VNavbar,
  ge as default
};
