// @gits-id/menu/dist/menu.mjs
import { defineComponent as k, toRefs as H, unref as e, openBlock as s, createElementBlock as d, renderSlot as S, createCommentVNode as h, computed as f, resolveComponent as I, createBlock as y, resolveDynamicComponent as $, normalizeClass as m, withCtx as i, createElementVNode as u, createVNode as g, toDisplayString as p, createTextVNode as w, Fragment as M, renderList as A } from "vue";
import { Icon as T } from "@iconify/vue";
import { VCollapsible as D } from "@gits-id/collapsible";
import { useRoute as E } from "vue-router";
const F = {
  key: 0,
  class: "menu-tooltip hidden shadow absolute bg-gray-100 text-gray-700 px-4 py-2.5 min-w-[200px] rounded left-20 sm:group-hover:block z-10"
}, C = /* @__PURE__ */ k({
  __name: "VMenuTooltip",
  props: {
    show: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:show"],
  setup(n) {
    const o = n, { show: t } = H(o);
    return (a, r) => e(t) ? (s(), d("span", F, [
      S(a.$slots, "default")
    ])) : h("", !0);
  }
}), q = /* @__PURE__ */ k({
  __name: "VMenuItem",
  props: {
    item: Object,
    mini: {
      type: Boolean,
      default: !1
    },
    textColor: {
      type: String,
      default: ""
    },
    dark: {
      type: Boolean,
      default: !1
    },
    expandHover: {
      type: Boolean,
      default: !1
    },
    small: {
      type: Boolean,
      default: !1
    },
    as: {
      type: String,
      default: ""
    }
  },
  setup(n) {
    const o = n, t = f(() => o.expandHover ? "justify-start sm:justify-center" : o.mini ? "justify-center flex justify-center" : ""), a = f(() => o.as || I("RouterLink"));
    return (r, v) => n.item ? (s(), y($(e(a)), {
      key: 0,
      to: n.item.to,
      exact: "",
      class: m(["group sub-menu gap-x-3 w-full px-2 py-2.5 rounded flex items-center truncate hover:bg-gray-100", [n.textColor, n.mini ? "py-4" : "", e(t)]])
    }, {
      default: i(() => [
        u("span", {
          class: m(n.mini ? "" : "px-1 w-6")
        }, [
          g(e(T), {
            icon: "ic:outline-circle",
            class: "w-4 h-4"
          })
        ], 2),
        u("span", {
          class: m({ "sm:hidden": n.mini && !n.expandHover, "text-sm": n.small })
        }, p(n.item.text), 3),
        g(C, {
          show: n.mini && !n.expandHover
        }, {
          default: i(() => [
            w(p(n.item.text), 1)
          ]),
          _: 1
        }, 8, ["show"])
      ]),
      _: 1
    }, 8, ["to", "class"])) : h("", !0);
  }
}), j = /* @__PURE__ */ k({
  __name: "VMenuIcon",
  props: {
    menu: null,
    icon: null
  },
  setup(n) {
    return (o, t) => typeof n.icon == "string" ? S(o.$slots, "icon", {
      key: 0,
      menu: n.menu
    }, () => [
      g(e(T), { icon: n.icon }, null, 8, ["icon"])
    ]) : (s(), y($(n.icon), {
      key: 1,
      class: "w-5 h-5"
    }));
  }
});
const G = (n, o) => {
  const t = n.__vccOpts || n;
  for (const [a, r] of o)
    t[a] = r;
  return t;
}, J = { class: "menu-icon w-6" }, K = ["src"], Q = ["innerHTML"], U = ["title"], W = ["src"], X = {
  inheritAttrs: !1
}, Y = /* @__PURE__ */ k({
  ...X,
  __name: "VMenu",
  props: {
    menu: {
      type: Object,
      default: null
    },
    mini: {
      type: Boolean,
      default: !1
    },
    expandHover: {
      type: Boolean,
      default: !1
    },
    dark: {
      type: Boolean,
      default: !1
    },
    color: {
      type: String,
      default: ""
    },
    bgColor: {
      type: String,
      default: ""
    },
    darkColor: {
      type: String,
      default: ""
    },
    darkBgColor: {
      type: String,
      default: ""
    },
    classMenuParent: {
      type: String,
      default: "text-primary-500 bg-gray-700"
    },
    small: {
      type: Boolean,
      default: !1
    }
  },
  setup(n) {
    const o = n, { menu: t, mini: a, dark: r, classMenuParent: v, expandHover: l } = H(o), b = f(() => r.value ? o.darkColor : o.color), L = f(() => r.value ? o.darkBgColor : o.bgColor), B = f(
      () => r.value ? "text-gray-400 hover:bg-gray-800 hover:text-gray-200" : "text-gray-700 hover:bg-gray-100"
    ), N = (x) => r.value ? x ? `${v.value}` : "" : x ? `${v.value}` : "", P = E(), V = (x) => {
      const _ = P?.path?.split("/");
      return x?.to?.split("/")?.[2] === _?.[2];
    }, R = f(() => [
      "font-normal hover:bg-gray-100 rounded !p-2 !py-3",
      o.mini ? "!justify-center" : ""
    ].join(" "));
    return (x, O) => {
      const _ = I("router-link");
      return e(t) ? (s(), d(M, { key: 0 }, [
        e(t).children ? (s(), y(e(D), {
          key: 0,
          "activator-class": e(R),
          "panel-class": "",
          "header-class": "",
          "hide-icon": e(a),
          "active-class": e(v)
        }, {
          header: i(({ isOpen: c }) => [
            u("div", {
              class: m(["flex items-center gap-x-3 rounded", [
                N(c),
                e(a) ? e(l) ? "" : "justify-center" : "",
                c ? e(L) : ""
              ]])
            }, [
              u("div", J, [
                e(t).img ? (s(), d("img", {
                  key: 0,
                  src: e(t).img,
                  alt: "img icon",
                  class: "w-5 h-5"
                }, null, 8, K)) : e(t).svg ? (s(), d("span", {
                  key: 1,
                  innerHTML: e(t).svg
                }, null, 8, Q)) : e(t).icon ? (s(), y(j, {
                  key: 2,
                  icon: e(t).icon,
                  menu: e(t)
                }, null, 8, ["icon", "menu"])) : h("", !0)
              ]),
              u("span", {
                title: e(t).text,
                class: m(["flex-grow text-left truncate", { "sm:hidden": e(a) && !e(l), "text-sm": n.small }])
              }, p(e(t).text), 11, U),
              g(C, {
                show: e(a) && !e(l)
              }, {
                default: i(() => [
                  w(p(e(t).text), 1)
                ]),
                _: 1
              }, 8, ["show"])
            ], 2)
          ]),
          default: i(() => [
            u("div", null, [
              (s(!0), d(M, null, A(e(t).children, (c, z) => (s(), y(q, {
                key: z,
                item: c,
                mini: e(a),
                expandHover: e(l),
                dark: e(r),
                small: c.small || n.small,
                "text-color": V(c) ? e(b) : e(B)
              }, null, 8, ["item", "mini", "expandHover", "dark", "small", "text-color"]))), 128))
            ])
          ]),
          _: 1
        }, 8, ["activator-class", "hide-icon", "active-class"])) : (s(), y(_, {
          key: 1,
          to: e(t).to,
          exact: "",
          class: m(["group menu-item transition duration-300 w-full px-2 py-3 rounded flex gap-x-4 items-center relative", [
            n.small ? "text-sm" : "",
            e(r) ? "hover:bg-gray-800" : "hover:bg-gray-100",
            V(e(t)) ? e(b) : e(B),
            e(a) && !e(l) ? "justify-start sm:justify-center" : ""
          ]])
        }, {
          default: i(() => [
            e(t).img ? (s(), d("img", {
              key: 0,
              src: e(t).img,
              alt: "img icon",
              class: "w-5 h-5"
            }, null, 8, W)) : h("", !0),
            g(j, {
              icon: e(t).icon,
              menu: e(t)
            }, null, 8, ["icon", "menu"]),
            u("span", {
              class: m({ "sm:hidden": e(a) && !e(l) })
            }, p(e(t).text), 3),
            g(C, {
              show: e(a) && !e(l)
            }, {
              default: i(() => [
                w(p(e(t).text), 1)
              ]),
              _: 1
            }, 8, ["show"])
          ]),
          _: 1
        }, 8, ["to", "class"]))
      ], 64)) : h("", !0);
    };
  }
}), oe = /* @__PURE__ */ G(Y, [["__scopeId", "data-v-598aa49e"]]);
export {
  oe as VMenu,
  j as VMenuIcon,
  q as VMenuItem,
  C as VMenuTooltip,
  oe as default
};
