// @gits-id/progress-bar/dist/progress-bar.mjs
import { computed as t, openBlock as c, createElementBlock as l, createElementVNode as s, normalizeStyle as u, normalizeClass as d, unref as i, renderSlot as f } from "vue";
const g = (e, r = "") => {
  switch (e) {
    case "primary":
      return "bg-primary-500";
    case "secondary":
      return "bg-secondary-500";
    case "info":
      return "bg-info-500";
    case "warning":
      return "bg-warning-500";
    case "success":
      return "bg-success-500";
    case "error":
      return "bg-error-500";
    case "default":
    default:
      return r || "bg-gray-600";
  }
}, p = { class: "relative pt-1" }, m = { class: "overflow-hidden h-2 mb-4 text-xs flex rounded bg-gray-200" }, y = {
  __name: "VProgressBar",
  props: {
    width: {
      type: Number,
      default: 50
    },
    color: {
      type: String,
      default: "default"
    }
  },
  setup(e) {
    const r = e, o = t(() => g(r.color)), a = t(() => [o.value]);
    return (n, h) => (c(), l("div", p, [
      s("div", m, [
        s("div", {
          style: u({ width: e.width + "%" }),
          class: d([i(a), "shadow-none flex flex-col text-center whitespace-nowrap text-white justify-center"])
        }, [
          f(n.$slots, "default")
        ], 6)
      ])
    ]));
  }
};
export {
  y as VProgressBar,
  y as default
};
